<!DOCTYPE html><html lang="en"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"><meta name="description" content=""><meta name="keywords" content=""><meta name="author" content="John Doe"><meta name="copyright" content="John Doe"><title>Yezi Blog</title><link rel="shortcut icon" href="/melody-favicon.ico"><link rel="stylesheet" href="/css/index.css?version=1.9.0"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/font-awesome@latest/css/font-awesome.min.css?version=1.9.0"><meta name="format-detection" content="telephone=no"><meta http-equiv="x-dns-prefetch-control" content="on"><link rel="dns-prefetch" href="https://cdn.jsdelivr.net"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><script>var GLOBAL_CONFIG = { 
  root: '/',
  algolia: undefined,
  localSearch: undefined,
  copy: {
    success: 'Copy successfully',
    error: 'Copy error',
    noSupport: 'The browser does not support'
  },
  hexoVersion: '6.0.0'
} </script><meta name="generator" content="Hexo 6.0.0"></head><body><i class="fa fa-arrow-right" id="toggle-sidebar" aria-hidden="true"></i><div id="sidebar" data-display="false"><div class="author-info"><div class="author-info__avatar text-center"><img src="/img/avatar.png"></div><div class="author-info__name text-center">John Doe</div><div class="author-info__description text-center"></div><hr><div class="author-info-articles"><a class="author-info-articles__archives article-meta" href="/archives"><span class="pull-left">Articles</span><span class="pull-right">6</span></a><a class="author-info-articles__categories article-meta" href="/categories"><span class="pull-left">Categories</span><span class="pull-right">1</span></a></div></div></div><nav class="no-bg" id="nav"><div id="page-header"><span class="pull-left"> <a id="site-name" href="/">Yezi Blog</a></span><i class="fa fa-bars toggle-menu pull-right" aria-hidden="true"></i><span class="pull-right"></span></div><div id="site-info"><div id="site-title">Yezi Blog</div><div id="site-sub-title"></div></div></nav><div id="content-outer"><div class="layout" id="content-inner"><div class="recent-post-item article-container"><a class="article-title" href="/2022/02/20/markdown%E8%AF%AD%E6%B3%95/">markdown语法</a><time class="post-meta__date"><i class="fa fa-calendar" aria-hidden="true"></i> 2022-02-20</time><span class="article-meta"><span class="article-meta__separator">|</span><i class="fa fa-inbox article-meta__icon" aria-hidden="true"></i><a class="article-meta__categories" href="/categories/uncategorized/">uncategorized</a></span><div class="content"><p>1.颜色<br><font color=red>他曾活过啊</font></p>
<p>```text<br><font face="黑体">我是黑体字</font><br><font face="微软雅黑">我是微软雅黑</font><br><font face="STCAIYUN">我是华文彩云</font><br><font color=red>我是红色</font><br><font color=#008000>我是绿色</font><br><font color=Blue>我是蓝色</font><br><font size=5>我是尺寸</font><br><font face="黑体" color=green size=5>我是黑体，绿色，尺寸为5</font><br>```</p>
<p><font face="黑体">我是黑体字</font><br><font face="微软雅黑">我是微软雅黑</font><br><font face="STCAIYUN">我是华文彩云</font><br><font color=red>我是红色</font><br><font color=#008000>我是绿色</font><br><font color=Blue>我是蓝色</font><br><font size=5>我是尺寸</font><br><font face="黑体" color=green size=5>我是黑体，绿色，尺寸为5</font></p>
<p>2.背景色</p>
<table><tr><td bgcolor=yellow>背景色yellow</td></tr></table>  

<p>3.居中  </p>
<center>居中</center>  
<p align="left">左对齐</p>  
<p align="right">右对齐</p>  
  
  

<p>4.下标<br>H<sub>2</sub>O</p>
<p>箭头</p>
<p>$H\stackrel{F}{\Leftrightarrow}$H<sub>2</sub>O</p>
<p>$X\stackrel{F}{\longrightarrow}Y$<br><a target="_blank" rel="noopener" href="https://blog.csdn.net/J__Max/article/details/86549124">https://blog.csdn.net/J__Max&#x2F;article&#x2F;details&#x2F;86549124</a></p>
</div><hr></div><div class="recent-post-item article-container"><a class="article-title" href="/2022/01/17/%E7%A7%91%E7%A0%94%E5%B9%B2%E8%B4%A7%20-%20FerrDb%20%E2%80%94%20%E9%93%81%E6%AD%BB%E4%BA%A1%E7%A0%94%E7%A9%B6%E5%88%86%E6%9E%90%E5%88%A9%E5%99%A8%EF%BC%81/">No title</a><time class="post-meta__date"><i class="fa fa-calendar" aria-hidden="true"></i> 2022-01-17</time><div class="content"><blockquote>
<p>本文由 <a target="_blank" rel="noopener" href="http://ksria.com/simpread/">简悦 SimpRead</a> 转码， 原文地址 <a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/405474253">zhuanlan.zhihu.com</a></p>
</blockquote>
<blockquote>
<p>铁死亡(Ferroptosis)是一种铁依赖性的调节性细胞死亡方式，与多种人类疾病（例如癌症、心血管疾病和退行性疾病等）紧密相关。铁死亡的过程很复杂，由多种代谢物和生物分子组成，也有一些特征基因发生变化。现如今…</p>
</blockquote>
<p>铁死亡 (Ferroptosis) 是一种铁依赖性的调节性细胞死亡方式，与多种人类疾病（例如癌症、心血管疾病和退行性疾病等）紧密相关。铁死亡的过程很复杂，由多种代谢物和生物分子组成，也有一些特征基因发生变化。现如今铁死亡已成为除 m6A、自噬、肠道微生物、肿瘤免疫等明星研究的又一大热点，开启这样一个新课题不知从何入手？FerrDb 数据库可以助你轻松入门铁死亡研究热点，达到事半功倍的效果！</p>
<p>FerrDb 数据库是由广州医科大学附属脑科医院、四川大学生命科学学院以及生物资源与生态环境教育部重点实验研究团队共同建成的， FerrDb 是现如今第一个人工整理的铁死亡数据库，用于管理和鉴定铁死亡相关的标志物和调控因子以及铁死亡关联疾病的数据库 。截止到目前为止，该数据库从 PubMed 数据库下载了 784 篇关于铁死亡的文章，并从中提取整理了 259 个调控基因、111 个标记和 95 个铁死亡相关疾病 。FerrDb 中的数据都可免费下载和使用。FerrDb 网址：<a href="https://link.zhihu.com/?target=http://www.zhounan.org/ferrdb">http://www.zhounan.org/ferrdb</a></p>
<p>先来一波视频演示吧</p>
<p><a target="_blank" rel="noopener" href="https://www.zhihu.com/zvideo/1416086019796561920">ferrDb 数据库实操演示<img src="https://pic1.zhimg.com/v2-cead3ee3c7b01eac5561a75f2d625537_s.jpg?source=12a79843">生物信息学培训的视频 · 947 播放</a></p>
<p><strong>一、网站主页</strong></p>
<p><img src="https://pic1.zhimg.com/v2-c15f1730928d895280e2b85cef89fa24_r.jpg"></p>
<p><strong>二、直接搜索基因</strong>  </p>
<p><img src="https://pic4.zhimg.com/v2-17a3698e1629e96e87ac5cc9ecbad477_r.jpg"></p>
<p>在检索框中，输入我们想讨研究的铁死亡基因，我们以 GPX4 为例。</p>
<p><img src="https://pic1.zhimg.com/v2-797535e0db9648f542c92c07e07f242c_r.jpg"></p>
<p>在结果页面我们可以看出在 Driver，Inducer, Inhibitor 和 Disease 分类中无结果，在 Suppressor 和 Marker 分类中检索到了结果，说明 GPX4 是铁死亡的抑制基因和标志物。结果中 GPX4 的相关信息包括 ID、验证方法、实验对象、作用途径、蛋白编号和参考文献等等。</p>
<p>点击 “View” 可以看到详细的基因铁死亡作用途径：</p>
<p><img src="https://pic2.zhimg.com/v2-9dd3caf41cdec2f406f30a7c31dd7a95_r.jpg"></p>
<p>点击 “UniProtKB” 蓝色链接可以直接进入 Uniport 网站查看该基因其他更多的详细信息 ：</p>
<p><img src="https://pic1.zhimg.com/v2-bc2e0dd1e88acb7ca1543fb1635159c0_r.jpg"></p>
<p>点击 “PMID” 可以直接链接到 pubmed 中的参考文献：</p>
<p><img src="https://pic3.zhimg.com/v2-b69dc8d67870c2ef3e528d985f6e10ea_r.jpg"></p>
<p><strong>三、六种分析模块</strong></p>
<p>在网站首页我们可以看到有六个研究模块，包括 Drivers ，Suppressors ，Markers ，Inducers， Inhibiters 和 Diseases。</p>
<ul>
<li>uDrivers 模块研究促进铁死亡的基因；</li>
<li>uSuppressors 模块研究</li>
<li>uMarkers 模块研究标记铁死亡发生的基因；</li>
<li>uInducers 模块研究可以引起铁死亡的小分子；</li>
<li>uInhibiters 模块研究可以抑制铁死亡的小分子；</li>
<li>uDiseases 模块研究与铁死亡相关的疾病和铁死亡在其中的作用。</li>
</ul>
<p>我们以 Driver 和 Disease 为例：</p>
<p><img src="https://pic1.zhimg.com/v2-e0c75dacb8e19bb70fe360247669ac18_r.jpg"></p>
<p>点击 Driver，进入如下页面：</p>
<p><img src="https://pic4.zhimg.com/v2-2bb17c69b2e917346344741d338ead7b_r.jpg"></p>
<p>这里展示的基因的信息与上述一致，包含 ID、验证方法、实验对象、作用途径、蛋白编号和参考文献等等，不再展开介绍。</p>
<p>点击 Disease，进入如下页面，显示与铁死亡相关的疾病，涉及的基因链接和文献链接：</p>
<p><img src="https://pic4.zhimg.com/v2-0367448e6dc72df3b14a2140823946d3_r.jpg"></p>
<p><strong>四、主菜单功能</strong></p>
<p>主菜单中可以直接找到在主页面直接链接的部分内容，另外，可以从主菜单中进入数据下载和基因互作网络展示等。</p>
<p><img src="https://pic4.zhimg.com/v2-51e3cdf4f437047eabe237502a206f2f_r.jpg"></p>
<p>在 Brower 下拉框中，前面的几项与主页面的链接内容是一样的，多了一项 Multirole regulator，点击后，出现如下表格：</p>
<p><img src="https://pic2.zhimg.com/v2-be11ea3b2fe01a6b2894c1fa7b72e861_r.jpg"><img src="https://pic2.zhimg.com/v2-bb0c18e1dfcb95260a76d916c52dcf65_r.jpg"></p>
<p>点击某一基因，进入基因信息页面，与直接搜索某基因的结果一致：</p>
<p><img src="https://pic1.zhimg.com/v2-5d112f9359780c8c8973ea1c3ac57594_r.jpg"></p>
<p>点击 “Download”，进入以下页面：在铁死亡相关文献中发现了 259 个基因，这些基因的注释揭示了 108 个驱动基因，69 个抑制基因和 111 个基因标志物。这三种基因之和是 288 个基因，与 259 个不符，是因为有多个基因属于多个注释集，如 venn 图所示。表格中也展示了调控铁死亡的小分子数量，及其中被标记为诱导剂和抑制剂的数量，铁死亡关联的疾病种类的数量等。在 Annotation data 中，这些数据可以以不同的文件形式下载。</p>
<p><img src="https://pic4.zhimg.com/v2-38e4837fa78ec79bfecf0e2dced5b337_r.jpg"></p>
<p>点击 “Utilities” 下拉框中的“regulation network”，进入以下页面：</p>
<p><img src="https://pic1.zhimg.com/v2-236b0d47284349f44dc3401497d153e0_r.jpg"></p>
<p>可以选择展示 Driver 或 Suppressor 的网络，是整合各个途径来生成的调控网络，并可以点击特定节点查看子网络：</p>
<p><img src="https://pic4.zhimg.com/v2-38aa1754c429359516fa8479f50d189f_r.jpg"></p>
<p>比如，我们点击 Suppressor 中的 TP53，可以生成 TP53 的调控子网络：</p>
<p><img src="https://pic4.zhimg.com/v2-addf4f7a64bbf88f64e7e71fe4adbbef_b.jpg"></p>
<p><strong>结语</strong></p>
<p>本次铁死亡数据库的使用就展示到这里，里面细节的部分，小伙伴们可以根据自己的需要进一步挖掘内容，一起实操起来吧！</p>
</div><hr></div><div class="recent-post-item article-container"><a class="article-title" href="/2022/01/17/TCGA%20%E4%B8%B4%E5%BA%8A%E6%95%B0%E6%8D%AE%E4%B8%8B%E8%BD%BD%E4%B8%8E%E6%95%B4%E7%90%86%20-%20%E7%94%9F%E4%BF%A1%E4%B9%8B%E5%AE%B6%20-%20%E7%94%9F%E4%BF%A1%E4%B9%8B%E5%AE%B6/">No title</a><time class="post-meta__date"><i class="fa fa-calendar" aria-hidden="true"></i> 2022-01-17</time><div class="content"><blockquote>
<p>本文由 <a target="_blank" rel="noopener" href="http://ksria.com/simpread/">简悦 SimpRead</a> 转码， 原文地址 <a target="_blank" rel="noopener" href="https://www.proteinstuff.com/8050.html">www.proteinstuff.com</a></p>
</blockquote>
<blockquote>
<h3 id="TCGA-临床数据下载与整理教程（代码-注释）！！！-…"><a href="#TCGA-临床数据下载与整理教程（代码-注释）！！！-…" class="headerlink" title="TCGA 临床数据下载与整理教程（代码 + 注释）！！！ ######################### […]"></a>TCGA 临床数据下载与整理教程（代码 + 注释）！！！ ######################### […]</h3></blockquote>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br></pre></td><td class="code"><pre><span class="line">### TCGA临床数据下载与整理教程（代码+注释）！！！</span><br><span class="line">###########################################################################################</span><br><span class="line">### 第一步：网页筛选下载（示例：肺腺癌LUAD）</span><br><span class="line"> ## 百度&quot;TCGA GDC&quot;，进入网址：https://portal.gdc.cancer.gov/</span><br><span class="line"> ## 进入官网，点击&quot;Repository&quot;，在左侧栏里选择&quot;Cases&quot;</span><br><span class="line"> ## 在&quot;Primary Site&quot;（原发位置）里点击选择&quot;bronchus and lung&quot;</span><br><span class="line"> ## 在&quot;Program&quot;（项目）里点击选择&quot;TCGA&quot;</span><br><span class="line"> ## 在&quot;Project&quot;里点击选择&quot;TCGA-LUAD&quot;这一亚型</span><br><span class="line"> ## 没有特殊研究目的的话，其它选项全部默认</span><br><span class="line"> ## 在左侧栏里选择&quot;Files&quot;</span><br><span class="line"> ## 在&quot;Data Category&quot;（数据目录）里点击选择&quot;clinical&quot;</span><br><span class="line"> ## 在&quot;Data Format&quot;（数据格式）里点击选择&quot;bcr xml&quot;</span><br><span class="line"> ## 点击&quot;Add All Files to Cart&quot;，添加进购物车（添加之前注意清空购物车）</span><br><span class="line"> ## 点击右上角购物车&quot;Cart&quot;</span><br><span class="line"> ## 下载Cart文件（打包文件）：点击&quot;Download&quot; - &quot;Cart&quot;</span><br><span class="line"> ## 这边下载的是xml文件，下面处理时需要用R包XML</span><br><span class="line"> ## 完成（1）</span><br><span class="line">### 也可以在&quot;Download&quot; - &quot;Cart&quot;页面左边的&quot;Clinical&quot;按钮处下载tsv格式的临床信息文件（2）</span><br><span class="line">### 注意：</span><br><span class="line"> ## tsv文件里的死亡时间和最后随访时间比xml文件里的新，所以做生存分析最好下载tsv文件！！！</span><br><span class="line"> ## xml文件里有些临床指标在tsv文件里没有，不过tsv文件里临床指标更多！！！</span><br><span class="line">### 这里再介绍使用国内Sangerbox软件下载xml格式临床信息文件的方法：</span><br><span class="line"> ## 点击&quot;TCGA简易下载工具&quot; - &quot;TCGA-LUAD&quot; - &quot;临床随访信息&quot; - &quot;Clinical BCR XML&quot; - 下载</span><br><span class="line"> ## 完成啦</span><br><span class="line">###########################################################################################</span><br><span class="line">### 第二步（方法1）：读取上面已下载好的tsv格式的临床信息文件</span><br><span class="line"> ## 注意修改路径！</span><br><span class="line">getwd()</span><br><span class="line">### 读取</span><br><span class="line">TSVClinicalDataPath  - &quot;/Users/yuanshi/Desktop/生信笔记/C_TCGA临床数据下载与整理/clinical.cart.2020-07-17/clinical.tsv&quot;</span><br><span class="line">FromeTSVClinData  - read.table(TSVClinicalDataPath,</span><br><span class="line">                               sep = &quot;t&quot;, header = T,</span><br><span class="line">                               fill = TRUE, quote = &quot;&quot;)</span><br><span class="line">### read.table里各项参数的意义，参考：https://oindrilasen.com/2018/01/error-in-scanfile-file-while-suing-read_table/</span><br><span class="line">### 提取部分我们需要的临床数据：</span><br><span class="line">partlyTSVClinData  - FromeTSVClinData[,c(&quot;case_submitter_id&quot;,</span><br><span class="line">                                         &quot;gender&quot;,</span><br><span class="line">                                         &quot;days_to_death&quot;,</span><br><span class="line">                                         &quot;days_to_last_follow_up&quot;,</span><br><span class="line">                                         &quot;vital_status&quot;,</span><br><span class="line">                                         &quot;ajcc_pathologic_stage&quot;,</span><br><span class="line">                                         &quot;ajcc_pathologic_t&quot;,</span><br><span class="line">                                         &quot;ajcc_pathologic_n&quot;,</span><br><span class="line">                                         &quot;ajcc_pathologic_m&quot;)]</span><br><span class="line">### 更换列名</span><br><span class="line">colnames(partlyTSVClinData)  - c(&quot;Barcode&quot;,&quot;Gender&quot;,&quot;DeathTime&quot;,</span><br><span class="line">                                 &quot;FollowTime&quot;,&quot;VitalStatus&quot;,&quot;Stage&quot;,</span><br><span class="line">                                 &quot;T_Stage&quot;,&quot;N_Stage&quot;,&quot;M_Stage&quot;)</span><br><span class="line">### 去重：我们想要的那一行为FALSE，所以要取反</span><br><span class="line">partlyTSVClinData  - partlyTSVClinData[!duplicated(partlyTSVClinData$Barcode),]</span><br><span class="line">### 保存一下</span><br><span class="line">partlyTSVClinDataFN  - &quot;partlyTSVClinData.csv&quot;</span><br><span class="line">write.csv(partlyTSVClinData,</span><br><span class="line">          file = partlyTSVClinDataFN,</span><br><span class="line">          row.names = F, quote = F)</span><br><span class="line">###########################################################################################</span><br><span class="line">### 第二步（方法2）：读取上面已下载好的xml格式的临床信息文件</span><br><span class="line">### 示例xml文件百度云链接：https://pan.baidu.com/s/1mSoTpbISeEQ4z1vzVcLnaA 密码：ot2k</span><br><span class="line">### 获取已下载的xml文件的地址并读取（整体压缩包需要先解压一下）</span><br><span class="line">getwd()</span><br><span class="line">XMLClinicalDataPath  - &quot;/Users/yuanshi/Desktop/生信笔记/C_TCGA临床数据下载与整理/XML&quot;</span><br><span class="line">partlyXMLClinDataFN  - &quot;partlyXMLClinData.csv&quot;</span><br><span class="line">#################</span><br><span class="line">### 数据处理！！！</span><br><span class="line">dir.create(&quot;tempClinXMLFiles&quot;)</span><br><span class="line">filepath  - dir(path = XMLClinicalDataPath,full.names = TRUE)</span><br><span class="line">### 将所有文件移动到同一个文件夹tempClinXMLFiles下</span><br><span class="line">### 使用以下循环：</span><br><span class="line">for(wd in filepath)&#123;</span><br><span class="line">  files  - dir(path = wd,pattern = &quot;xml$&quot;) # 查看满足条件(pattern=xml)的文件</span><br><span class="line">  fromfilepath  - paste(wd,&quot;/&quot;,files,sep = &quot;&quot;)</span><br><span class="line">  tofilepath  - paste(&quot;./tempClinXMLFiles/&quot;,files,sep = &quot;&quot;)</span><br><span class="line">  file.copy(fromfilepath,tofilepath)</span><br><span class="line">#################</span><br><span class="line">### 临床数据整合！！！</span><br><span class="line">setwd(&quot;./tempClinXMLFiles&quot;)</span><br><span class="line">xmlFileNames  - dir(path = &quot;./&quot;, pattern = &quot;xml$&quot;) # list.files()函数也可以</span><br><span class="line">library(XML)</span><br><span class="line">library(methods)</span><br><span class="line">FromeXMLClinData  - lapply(xmlFileNames,</span><br><span class="line">                           function(x)&#123;            # x = xmlFileNames[1]</span><br><span class="line">                           result  - xmlParse(file = x)</span><br><span class="line">                           rootnode  - xmlRoot(result) </span><br><span class="line">                           xmldataframe  - xmlToDataFrame(rootnode[2])</span><br><span class="line">                           return(t(xmldataframe))&#125;)</span><br><span class="line">### 上面这段代码不错</span><br><span class="line">FromeXMLClinData  - t(do.call(cbind, FromeXMLClinData))</span><br><span class="line">### 返回上级目录，保存一下</span><br><span class="line">setwd(&quot;../&quot;)</span><br><span class="line">write.csv(FromeXMLClinData, file = &quot;FromeXMLClinData.csv&quot;)</span><br><span class="line">### 转换成数据框格式</span><br><span class="line">FromeXMLClinData  - as.data.frame(FromeXMLClinData)</span><br><span class="line">### 提取部分我们需要的临床数据：</span><br><span class="line">partlyXMLClinData  - data.frame(Barcode = FromeXMLClinData$bcr_patient_barcode,</span><br><span class="line">                                Vital_status = FromeXMLClinData$vital_status,</span><br><span class="line">                                days_to_death = FromeXMLClinData$days_to_death,</span><br><span class="line">                                lastFollowupTime = FromeXMLClinData$days_to_last_followup,</span><br><span class="line">                                Stage = FromeXMLClinData$stage_event)</span><br><span class="line">### 保存一下</span><br><span class="line">write.csv(partlyXMLClinData,</span><br><span class="line">          file = partlyXMLClinDataFN,</span><br><span class="line">          row.names = F, quote = F)</span><br><span class="line">### 完结撒花！！！</span><br><span class="line">### 完结撒花！！！</span><br><span class="line">### 完结撒花！！！</span><br><span class="line"></span><br></pre></td></tr></table></figure></div><hr></div><div class="recent-post-item article-container"><a class="article-title" href="/2022/01/16/R%20%E8%AF%AD%E8%A8%80%20gzfile%20%E4%B8%8D%E8%A7%A3%E5%8E%8B%E7%9B%B4%E6%8E%A5%E8%AF%BB%E5%8F%96%E5%8E%8B%E7%BC%A9%E6%96%87%E6%9C%AC%20-%20%E7%AE%80%E4%B9%A6/">No title</a><time class="post-meta__date"><i class="fa fa-calendar" aria-hidden="true"></i> 2022-01-16</time><div class="content"><blockquote>
<p>本文由 <a target="_blank" rel="noopener" href="http://ksria.com/simpread/">简悦 SimpRead</a> 转码， 原文地址 <a target="_blank" rel="noopener" href="https://www.jianshu.com/p/3c737c5139ce">www.jianshu.com</a></p>
</blockquote>
<blockquote>
<p>R语言是常用的数据处理语言，对于读取文本文件，很多时候如果对文本进行压缩，会节省很多空间，那么对于压缩的文本文件，能够直接在R中不解压，而直接读取其中的文本呢？ R的基础包中…</p>
</blockquote>
<p>R 语言是常用的数据处理语言，对于读取文本文件，很多时候如果对文本进行压缩，会节省很多空间，那么对于压缩的文本文件，能够直接在 R 中不解压，而直接读取其中的文本呢？</p>
<p>R 的基础包中就有专门处理这些的命令，包括函数 <strong>gzfile</strong>，<strong>bzfile</strong>，<strong>xzfile</strong> 和 <strong>unz</strong> 等，能够处理常见的 <strong>gzip</strong>、<strong>bgzip</strong>、<strong>bgzip2</strong>，<strong>zip</strong> 等压缩格式的文本。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">#实例文件</span><br><span class="line">a = data.frame(x = 1:10, y = paste(&quot;n&quot;, 1:10, sep =&quot;&quot;) )</span><br><span class="line">#将a写入文件中</span><br><span class="line">write.table(a, file = &quot;test.txt&quot;, row.names = F)</span><br><span class="line">#压缩文件</span><br><span class="line">system(&quot;gzip test.txt&quot;)</span><br><span class="line">#读取压缩文件</span><br><span class="line">a1 = read.table( gzfile(&quot;test.txt.gz&quot;), header = T)</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure></div><hr></div><div class="recent-post-item article-container"><a class="article-title" href="/2022/01/15/%E5%A6%82%E4%BD%95%E4%B8%8B%E8%BD%BD%20TCGA%20%E6%95%B0%E6%8D%AE%20-%20TCGA%20%E6%95%B0%E6%8D%AE%E4%B8%8B%E8%BD%BD%E5%B0%8F%E6%95%99%E7%A8%8B%20-%20%E7%AE%80%E4%B9%A6/">No title</a><time class="post-meta__date"><i class="fa fa-calendar" aria-hidden="true"></i> 2022-01-15</time><div class="content"><blockquote>
<p>本文由 <a target="_blank" rel="noopener" href="http://ksria.com/simpread/">简悦 SimpRead</a> 转码， 原文地址 <a target="_blank" rel="noopener" href="https://www.jianshu.com/p/5b90c05e5101">www.jianshu.com</a></p>
</blockquote>
<blockquote>
<ol>
<li>什么是TCGA？TCGA中有哪些数据？ TCGA的全称是The Cancer Genome Atlas, 这个项目始于2005年，它旨在使用基因测序和生物信息学编目与癌…</li>
</ol>
</blockquote>
<h4 id="1-什么是-TCGA？TCGA-中有哪些数据？"><a href="#1-什么是-TCGA？TCGA-中有哪些数据？" class="headerlink" title="1. 什么是 TCGA？TCGA 中有哪些数据？"></a>1. 什么是 TCGA？TCGA 中有哪些数据？</h4><blockquote>
<p>TCGA 的全称是 The Cancer Genome Atlas, 这个项目始于 2005 年，它旨在使用基因测序和生物信息学编目与癌症有关的基因突变。TCGA 通过利用高通量基因组分析技术，来帮助我们更好地理解癌症的遗传学基础，从而提升我们对于癌症的诊断能力和对癌症的治疗、预防。<br>TCGA 受美国癌症研究所 (National Cancer Institute) 下的癌症基因组中心和美国人类基因组研究所监管。<br>TCGA 包括主要进行测序的基因组表征中心 (genome characterization centers, GCCs) 和负责测序数据分析的基因组数据分析中心(genome data analysis centers, GDACs)，到目前为止 TCGA 共有 39 种癌症的相关测序数据，涉及 29 种癌症器官，1 万多个肿瘤样本，27 万多份文件。</p>
</blockquote>
<p><strong>• 那么可以从 TCGA 中下载到哪些类型的数据呢？</strong><br>TCGA 的数据类型主要有以下几种：<br>(1) <strong>Clinical</strong>: 包括病人的一般情况、诊治情况、TNM 分期、肿瘤病理、生存情况等。<br>(2) <strong>mRNA 表达数据</strong>: 通过 mRNA 芯片或者 RNAseq 测得的 mRNA 表达量<br>(3)<strong>microRNA</strong>: microRNA 芯片或者 microRNA-Seq 测得的 microRNA 表达量<br>(4) <strong>Copy number variation</strong>: SNP 芯片得到的肿瘤组织比对正常组织的染色体上各片段的比值<br>(5) <strong>Mutation</strong>: 肿瘤组织测序结果相对参考基因组的核苷酸突变，包括插入和缺失等变化<br>(6) <strong>Protein</strong>: 蛋白芯片测序得到的约 200 种常见癌症相关蛋白的表达量<br>(7) <strong>Mythelation</strong>: 甲基化芯片测得的 DNA 甲基化数据，主要为 27 和 450 两种芯片的数据</p>
<p>其中 mRNA-Seq，miRNA-Seq 以及 Methylation Array 被广泛使用。</p>
<blockquote>
<p><strong>mRNA-Seq 数据分为 3 种</strong>：<br>HTSeq-Counts；HTSeq-FPKM；HTSeq-FPKM-UQ。<br>前两个比较好理解，第三个跟第二个的区别在于不同的标准化方法，公式可参考 <a target="_blank" rel="noopener" href="https://links.jianshu.com/go?to=https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/">https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines&#x2F;Expression_mRNA_Pipeline&#x2F;</a></p>
</blockquote>
<p><strong>• TCGA 数据等级：</strong></p>
<p>level1：原始数据<br>level2：处理过的数据<br>level3：经过分割、解释的数据<br>level4：感兴趣的区域或概要<br>总而言之，前面 2 个层级的数据一般是拿不到的，需要权限，一般也只有国外的 PI 才能申请到（听说的），我们一般拿到的 open 数据就属于那种已经标准化后的数据。</p>
<p><strong>• TCGA 样本分类：</strong><br>除了要知道数据等级外，我们还需要了解 TCGA 的样本分类，比如哪个是正常样本，哪个是肿瘤样本</p>
<p>一般我们可以看到样品名称如：TCGA-19-2619-10A，我们需要关注的是最后一位 10A，一般来说 01 代表癌症样本，11 代表癌旁样本。其实从 01-09 是 tumor，癌症样本；10-29 是 normal，癌旁样本。只是其中分的比较细</p>
<p>具体可参考官网说明：<a target="_blank" rel="noopener" href="https://links.jianshu.com/go?to=https://wiki.nci.nih.gov/display/TCGA/TCGA+barcode">https://wiki.nci.nih.gov/display/TCGA/TCGA+barcode</a></p>
<h4 id="2-TCGA-数据下载方式"><a href="#2-TCGA-数据下载方式" class="headerlink" title="2. TCGA 数据下载方式"></a>2. TCGA 数据下载方式</h4><blockquote>
<p>主要有三种方式可以下载 TCGA 的数据，一是利用 GDC 官方的下载工具；二是利用 cbioportal 下载；三是 TCGA-assembler 2。</p>
</blockquote>
<p>在这里我将着重分享一下<strong>利用 GDC 进行 TCGA 数据下载的方式</strong>。</p>
<h5 id="1-GDC-官方下载工具下载和安装"><a href="#1-GDC-官方下载工具下载和安装" class="headerlink" title="(1) GDC 官方下载工具下载和安装"></a>(1) GDC 官方下载工具下载和安装</h5><p>TCGA GDC Data Portal 官网地址为：<a target="_blank" rel="noopener" href="https://links.jianshu.com/go?to=https://portal.gdc.cancer.gov/">https://portal.gdc.cancer.gov/</a></p>
<p>进入 GDC 主页面之后，选择 “Repository” 进行查看以及下载数据 (Browse and download data)；</p>
<p><img src="http://upload-images.jianshu.io/upload_images/16163460-aaa822799af1dc30.png"> GDC home page</p>
<p>进来 “Reoository” 的界面如下图。GDC 给出了一系列的用户友好的选择框，你只需要根据条条框框来选择就可以下载到自己想要的数据，而不需要去几百个文件夹里面漫无目的的查找了。根据自己课题的需求，在对应的 “Cases” 框和 “Files” 分别选中你要的条件即可。</p>
<p><img src="http://upload-images.jianshu.io/upload_images/16163460-03f0591f0397820c.png"> Repository 界面</p>
<p>一般在下载数据时先设定 “Cases” 的条件，再设置 “Files” 的条件。下面以下载乳腺癌 miRNA 表达数据为例进行演示：<br><strong>• 在 Cases 界面分别对 Primary Site, Project, Disease Type 几个主要信息进行设置</strong>，如下图：  </p>
<p><img src="http://upload-images.jianshu.io/upload_images/16163460-59e6ac8465dbdb76.png"> TCGA 数据下载示例 1</p>
<p><strong>• 在 Files 界面进行选择，它包括 Data Category, Data Type, Exprimental Strategy, Workflow Type 等</strong></p>
<p>。选择如下图：</p>
<p><img src="http://upload-images.jianshu.io/upload_images/16163460-d96cb7c25a6586a7.png"> TCGA 数据下载示例 2</p>
<p><strong>• Cases 和 Files 都选择好了之后，点击 “Manifest” 下载相应的. manifest 文件</strong> <img src="http://upload-images.jianshu.io/upload_images/16163460-109f0939be7e9031.png"> TCGA 数据下载示例 3</p>
<p><strong>• 下载选择这批 Cases 的 clinical 信息文件：</strong></p>
<p>即不改变 Cases 部分的选择，清空 Files 的勾选项，在 Data Category 中选择 Clinical，再在 Data FOrmat 中选择 XML 格式，如下图：</p>
<p><img src="http://upload-images.jianshu.io/upload_images/16163460-fc8c46b939b8b9c1.png"> TCGA 数据下载示例 4</p>
<p>同样的选择好后，下载 manifest 文件。</p>
<p><em><strong>▷ Tip: 注意此时得到的样本数量，这在后续用下载得到的 miRNA 表达数据结合临床信息进行生存分析的时候有用</strong></em></p>
<p><strong>• 两个 manifest 文件下载完后，通过 Filezilla Client 这个软件将本地的 manifext 相关的两个. txt 后缀的文件传输到 linux 服务器上</strong><br>(这一步是否要进行取决于你想在哪边使用 GDC client 这个软件，由于我之后会直接把 GDC client 下载到 linux 服务器上，所以同时也把 manifest.txt 上传到了服务器)<br>_<strong>▷附 GDC Data Transfer Toll Client 地址</strong>_：<a target="_blank" rel="noopener" href="https://links.jianshu.com/go?to=https://gdc.cancer.gov/access-data/gdc-data-transfer-tool">https://gdc.cancer.gov/access-data/gdc-data-transfer-tool</a></p>
<p><img src="http://upload-images.jianshu.io/upload_images/16163460-4a756534eab7f2c6.png"> TCGA 数据下载示例 5</p>
<blockquote>
<p>接下来主要是 Linux 服务器上的命令相关操作</p>
</blockquote>
<h6 id="1-Linux-环境下下载并解压-GDC-Client"><a href="#1-Linux-环境下下载并解压-GDC-Client" class="headerlink" title="1. Linux 环境下下载并解压 GDC Client"></a>1. Linux 环境下下载并解压 GDC Client</h6><p><code>wget https://gdc.cancer.gov/system/files/authenticated%20user/0/gdc-client_v1.4.0_Ubuntu_x64.zip</code><br><code>unzip gdc-client_v1.4.0_Ubuntu_x64.zip</code></p>
<h6 id="2-利用-GDC-Client-下载-menifest-txt-文件中的数据"><a href="#2-利用-GDC-Client-下载-menifest-txt-文件中的数据" class="headerlink" title="2. 利用 GDC Client 下载 menifest.txt 文件中的数据"></a>2. 利用 GDC Client 下载 menifest.txt 文件中的数据</h6><p><code>./gdc-client --help</code><br><code>./gdc-client download --help</code><br><code>./gdc-client download -m gdc_manifest.2019_04_24_BRCA_clinical.txt</code><br>同理根据 miRNA 的 manifest 文件下载 miRNA 的数据。</p>
<h6 id="3-查看样本的生存情况"><a href="#3-查看样本的生存情况" class="headerlink" title="3. 查看样本的生存情况"></a>3. 查看样本的生存情况</h6><p><code>grep -i vital_status */*xml | grep -v Alive | wc</code><br>因为 false 不代表 death，所以这里选择查看的是 alive 的样本数量。<br>当你不那么会编程，但是有需要整理下载下来这些样本时，你可能会需要以下命令：<br><code>grep -i vital_status */*xml|grep Alive |cut -d&quot;.&quot; -f 3|sort -u |wc</code></p>
<p>Reference:<br>[1] <a target="_blank" rel="noopener" href="https://links.jianshu.com/go?to=https://en.wikipedia.org/wiki/The_Cancer_Genome_Atlas">TCGA-wikipedia</a><br>[2] <a target="_blank" rel="noopener" href="https://links.jianshu.com/go?to=%255Bhttp://www.bioinfo-scrounger.com/archives/271%255D(http://www.bioinfo-scrounger.com/archives/271)">初步认识 TCGA</a><br>[3] <a target="_blank" rel="noopener" href="https://links.jianshu.com/go?to=https://share.weiyun.com/53OUgRO?from=singlemessage&isappinstalled=0">生信技能树 - TCGA 肿瘤数据库知识图谱视频教程</a></p>
</div><hr></div><div class="recent-post-item article-container"><a class="article-title" href="/2022/01/15/04.mRNAmerge%20%E4%B8%AD.%20pl%20%E4%BB%A3%E7%A0%81%E4%B9%B1%E7%A0%81%E9%97%AE%E9%A2%98%20%C2%B7%20%E8%AF%AD%E9%9B%80/">No title</a><time class="post-meta__date"><i class="fa fa-calendar" aria-hidden="true"></i> 2022-01-15</time><div class="content"><blockquote>
<p>本文由 <a target="_blank" rel="noopener" href="http://ksria.com/simpread/">简悦 SimpRead</a> 转码， 原文地址 <a target="_blank" rel="noopener" href="https://www.yuque.com/docs/share/c0ae1397-6f6c-4a3a-b0c1-84908afea84d">www.yuque.com</a></p>
</blockquote>
<blockquote>
<p>前言目前因为代码加密的原因，诸如 merge.pl 和 mo…</p>
</blockquote>
<p><img src="https://gw.alipayobjects.com/zos/rmsportal/XuVpGqBFxXplzvLjJBZB.svg"></p>
<h5 id="Join-Yuque-to-read-better"><a href="#Join-Yuque-to-read-better" class="headerlink" title="Join Yuque to read better"></a>Join Yuque to read better</h5><p>Want to follow this author to read more or favorite this article? Please or to Yuque</p>
<p>前言<br>目前因为代码加密的原因，诸如 merge.pl 和 moveFiles.pl 都存在乱码问题，本文主要利用 R 语言来实现 TCGA 样本的合并与注释操作  </p>
<p><img src="https://cdn.nlark.com/yuque/0/2022/png/21544045/1641091661473-997d949a-d37d-409a-8e5e-4e03a59de271.png?x-oss-process=image/resize,w_778,limit_0"></p>
<p>一、使用 R 语言实现 moveFiles.pl 的操作<br>1.1 数据来源<br>本步骤主要利用 metadata 和 Cart 的数据，本文以 TCGA-LAML 为例，共有 151 例样本  </p>
<p><img src="https://cdn.nlark.com/yuque/0/2022/png/21544045/1641091958593-b22020ba-ed2e-41b9-90ca-95982eaae8fe.png"></p>
<p><img src="https://cdn.nlark.com/yuque/0/2022/png/21544045/1641092283493-956fce05-5af6-4a35-b3a4-a82501aea8d9.png"></p>
<p>1.2 实际步骤<br>1.2.1 解压 gdc 压缩包<br>得到 151 个文件夹和 1 个 MANIFEST.txt，每个文件夹下是各样本的测序结果  </p>
<p><img src="https://cdn.nlark.com/yuque/0/2022/png/21544045/1641092625313-a20e9446-1e3f-4780-bd1c-e9cadec2f726.png"></p>
<p>1.2.2 moveFile  </p>
<ol>
<li>在文件检索框中搜索 “.gz”，将所有结果复制 &#x2F; 剪切到一个统一的文件夹 all_file 中。</li>
</ol>
<p><img src="https://cdn.nlark.com/yuque/0/2022/png/21544045/1641092859399-72c72169-e599-4e60-97bd-9555c7abd9fd.png?x-oss-process=image/resize,w_746,limit_0"></p>
<p><img src="https://cdn.nlark.com/yuque/0/2022/png/21544045/1641092899318-a213b0ad-5c78-4134-b079-81dff65a2140.png?x-oss-process=image/resize,w_460,limit_0"></p>
<p>所有结果复制 &#x2F; 剪切到一个统一的文件夹 all_file 中</p>
<p>1.2.3 合并表达谱矩阵<br>将第一步下载的 json 文件移动进 all_file 中，运行 TCGA_merge.R，得到 symbol.txt，代码中已进行 ID 转换，因此可跳过 05.idTrans 的操作  </p>
<p><img src="https://cdn.nlark.com/yuque/0/2022/png/21544045/1641093368053-d88d8215-a0cb-4ed2-8d74-94c5c8fe21ca.png"></p>
<p><img src="https://cdn.nlark.com/yuque/0/2022/png/21544045/1641093427560-092005ed-8014-444b-9fab-4b536f24047a.png"></p>
<p>二、TCGA_merge.R（需修改第一行）<br>​</p>
<figure class="highlight plaintext"><figcaption><span># 需修改</span></figcaption><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br></pre></td><td class="code"><pre><span class="line"># install.packages(c(&quot;rjson&quot;,&quot;jsonio&quot;,&quot;jsonlite&quot;)) #未安装包则先安装</span><br><span class="line">library(rjson)</span><br><span class="line">library(jsonlite)</span><br><span class="line">library(stringr)</span><br><span class="line">library(biomaRt)</span><br><span class="line">library(dplyr)</span><br><span class="line">rm(list = ls())</span><br><span class="line"></span><br><span class="line">json &lt;- fromJSON(txt = list.files(pattern = &quot;.json&quot;))</span><br><span class="line">MMRF_ID&lt;-list()</span><br><span class="line">matrix_MMRF&lt;-list()</span><br><span class="line">filename_in_json&lt;-list()</span><br><span class="line">for (i in 1:nrow(json)) &#123;</span><br><span class="line">  MMRF_ID[i]&lt;-json[[3]][[i]]</span><br><span class="line">&#125;</span><br><span class="line">MMRF_ID&lt;-do.call(rbind, MMRF_ID)</span><br><span class="line">for (i in 1:nrow(json)) &#123;</span><br><span class="line">  filename_in_json[i]&lt;-json[i,4]</span><br><span class="line">&#125;</span><br><span class="line">filename_in_json&lt;-do.call(rbind, filename_in_json)</span><br><span class="line">filename_in_json&lt;-as.character(filename_in_json)</span><br><span class="line">for (i in 1:length(filename_in_json)) &#123;</span><br><span class="line">  matrix_MMRF[[i]] = read.table( gzfile(filename_in_json[i]), header = F)[,2]</span><br><span class="line">&#125;</span><br><span class="line">probematrix&lt;-read.table( gzfile(filename_in_json[1]), header = F)[,1]</span><br><span class="line"></span><br><span class="line">matrix_MMRF&lt;-do.call(cbind, matrix_MMRF)</span><br><span class="line">matrix_MMRF&lt;-cbind(probematrix,matrix_MMRF)</span><br><span class="line"></span><br><span class="line">#名称修改</span><br><span class="line">MM_col&lt;-str_replace(MMRF_ID[,1],pattern = &quot;aq-&quot;,replacement = &quot;&quot;)</span><br><span class="line">MM_col&lt;-str_replace(MM_col,pattern = &quot;R&quot;,replacement = &quot;&quot;)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">colnames(matrix_MMRF)&lt;-c(&quot;ID&quot;,str_sub(MMRF_ID[,1],1,12))</span><br><span class="line"># CBF&lt;-read.table(&quot;CBF-b.txt&quot;,sep =&quot;\t&quot;,header = F)</span><br><span class="line"># clinical&lt;-read.table(&quot;sample with clinical.txt&quot;,sep =&quot;\t&quot;,header = F)</span><br><span class="line"># intersect(clinical$V1,colnames(matrix_MMRF))</span><br><span class="line"></span><br><span class="line">#ID转换</span><br><span class="line">#matrix_MMRF&lt;-read.table(&quot;symbol.txt&quot;,sep = &quot;\t&quot;,header = T,check.names = F)</span><br><span class="line">matrix_MMRF[,1]&lt;-str_remove(matrix_MMRF[,1],pattern = &quot;[.][1-9]&quot;)</span><br><span class="line">ensembl = useMart(biomart=&quot;ensembl&quot;,dataset=&quot;hsapiens_gene_ensembl&quot;)</span><br><span class="line">geneENS&lt;- getBM(attributes=c(&quot;ensembl_gene_id&quot;,&quot;hgnc_symbol&quot;),</span><br><span class="line">                filters = &quot;ensembl_gene_id&quot;,</span><br><span class="line">                values=matrix_MMRF[,1],</span><br><span class="line">                mart=ensembl)</span><br><span class="line">all&lt;-merge(geneENS,matrix_MMRF,by.x = &quot;ensembl_gene_id&quot;,by.y = &quot;ID&quot;)</span><br><span class="line">all&lt;-all[,-1]</span><br><span class="line">colnames(all)[1]&lt;-&quot;genes&quot;</span><br><span class="line"></span><br><span class="line">write.table(all,&quot;symbol.txt&quot;,sep = &quot;\t&quot;,row.names = F,col.names = T)</span><br></pre></td></tr></table></figure>
<p>​</p>
<p>If you get gains，please give a like</p>
<p>01-02 11:20</p>
<p>4</p>
<p>0</p>
<p>Content Report</p>
</div><hr></div><nav id="pagination"><div class="pagination"><span class="page-number current">1</span></div></nav></div></div><footer><div class="layout" id="footer"><div class="copyright">&copy;2013 - 2022 By John Doe</div><div class="framework-info"><span>Driven - </span><a target="_blank" rel="noopener" href="http://hexo.io"><span>Hexo</span></a><span class="footer-separator">|</span><span>Theme - </span><a target="_blank" rel="noopener" href="https://github.com/Molunerfinn/hexo-theme-melody"><span>Melody</span></a></div><div class="busuanzi"><script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script><span id="busuanzi_container_site_uv"><i class="fa fa-user"></i><span id="busuanzi_value_site_uv"></span><span></span></span><span class="footer-separator">|</span><span id="busuanzi_container_site_pv"><i class="fa fa-eye"></i><span id="busuanzi_value_site_pv"></span><span></span></span></div></div></footer><i class="fa fa-arrow-up" id="go-up" aria-hidden="true"></i><script src="https://cdn.jsdelivr.net/npm/animejs@latest/anime.min.js"></script><script src="https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.js"></script><script src="https://cdn.jsdelivr.net/npm/velocity-animate@latest/velocity.min.js"></script><script src="https://cdn.jsdelivr.net/npm/velocity-ui-pack@latest/velocity.ui.min.js"></script><script src="/js/utils.js?version=1.9.0"></script><script src="/js/fancybox.js?version=1.9.0"></script><script src="/js/sidebar.js?version=1.9.0"></script><script src="/js/copy.js?version=1.9.0"></script><script src="/js/fireworks.js?version=1.9.0"></script><script src="/js/transition.js?version=1.9.0"></script><script src="/js/scroll.js?version=1.9.0"></script><script src="/js/head.js?version=1.9.0"></script><script>if(/Android|webOS|iPhone|iPod|iPad|BlackBerry/i.test(navigator.userAgent)) {
  $('#nav').addClass('is-mobile')
  $('footer').addClass('is-mobile')
  $('#top-container').addClass('is-mobile')
}</script></body></html>