<!DOCTYPE html><html lang="en"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"><meta name="description" content="复现论文:铁死亡相关基因对肝细胞癌患者诊断及预后预测价值的研究"><meta name="keywords" content="Iron death"><meta name="author" content="Yezi"><meta name="copyright" content="Yezi"><title>复现论文:铁死亡相关基因对肝细胞癌患者诊断及预后预测价值的研究 | Yezi Blog</title><link rel="shortcut icon" href="/melody-favicon.ico"><link rel="stylesheet" href="/css/index.css?version=1.9.0"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/font-awesome@latest/css/font-awesome.min.css?version=1.9.0"><meta name="format-detection" content="telephone=no"><meta http-equiv="x-dns-prefetch-control" content="on"><link rel="dns-prefetch" href="https://cdn.jsdelivr.net"><link rel="dns-prefetch" href="https://hm.baidu.com"><script>var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "https://hm.baidu.com/hm.js?2c078d16d56c81e15f58694fc4b73a5e";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();</script><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><script src="https://v1.hitokoto.cn/?encode=js&amp;charset=utf-8&amp;select=.footer_custom_text" defer></script><script>var GLOBAL_CONFIG = { 
  root: '/',
  algolia: undefined,
  localSearch: undefined,
  copy: {
    success: 'Copy successfully',
    error: 'Copy error',
    noSupport: 'The browser does not support'
  },
  hexoVersion: '6.0.0'
} </script><meta name="generator" content="Hexo 6.0.0"></head><body><i class="fa fa-arrow-right" id="toggle-sidebar" aria-hidden="true"></i><div id="sidebar" data-display="true"><div class="toggle-sidebar-info text-center"><span data-toggle="Toggle article">Toggle site</span><hr></div><div class="sidebar-toc"><div class="sidebar-toc__title">Catalog</div><div class="sidebar-toc__progress"><span class="progress-notice">You've read</span><span class="progress-num">0</span><span class="progress-percentage">%</span><div class="sidebar-toc__progress-bar"></div></div><div class="sidebar-toc__content"><ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#1-%E4%B8%80%E8%88%AC%E8%B5%84%E6%96%99"><span class="toc-number">1.</span> <span class="toc-text">1. 一般资料</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#1-1-tcga%E8%BD%AC%E5%BD%95%E7%BB%84%E6%B5%8B%E5%BA%8F%E6%95%B0%E6%8D%AE%E4%B8%8B%E8%BD%BD"><span class="toc-number">1.0.1.</span> <span class="toc-text">1.1 tcga转录组测序数据下载</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#1-2-tcga%E4%B8%B4%E5%BA%8A%E8%B5%84%E6%96%99%E4%B8%8B%E8%BD%BD"><span class="toc-number">1.0.2.</span> <span class="toc-text">1.2 tcga临床资料下载</span></a></li></ol></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2-1-%E9%93%81%E6%AD%BB%E4%BA%A1%E7%9B%B8%E5%85%B3%E5%B7%AE%E5%BC%82%E8%A1%A8%E8%BE%BE%E5%9F%BA%E5%9B%A0%E7%9A%84%E7%AD%9B%E9%80%89"><span class="toc-number">2.</span> <span class="toc-text">2.1 铁死亡相关差异表达基因的筛选</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#2-1-%E7%AD%9B%E9%80%89%E5%B7%AE%E5%BC%82%E5%9F%BA%E5%9B%A0"><span class="toc-number">2.1.</span> <span class="toc-text">2.1 筛选差异基因</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#2-1-2-%E7%81%AB%E5%B1%B1%E5%9B%BE"><span class="toc-number">2.2.</span> <span class="toc-text">2.1.2 火山图</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#2-1-3-Venn%E5%9B%BE"><span class="toc-number">2.3.</span> <span class="toc-text">2.1.3 Venn图</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2-2-%E6%A8%A1%E5%9D%97%E5%9F%BA%E5%9B%A0%E7%AD%9B%E9%80%89"><span class="toc-number">3.</span> <span class="toc-text">2.2 模块基因筛选</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2-3-%E7%94%9F%E5%AD%98%E7%9B%B8%E5%85%B3%E6%A8%A1%E5%9D%97%E5%9F%BA%E5%9B%A0%E7%AD%9B%E9%80%89"><span class="toc-number">4.</span> <span class="toc-text">2.3 生存相关模块基因筛选</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2-4-%E7%94%9F%E5%AD%98%E7%9B%B8%E5%85%B3%E6%A8%A1%E5%9D%97%E5%9F%BA%E5%9B%A0%E7%9A%84%E5%8A%9F%E8%83%BD%E5%AF%8C%E9%9B%86%E5%88%86%E6%9E%90"><span class="toc-number">5.</span> <span class="toc-text">2.4 生存相关模块基因的功能富集分析</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2-5-%E7%AD%9B%E9%80%89%E7%8B%AC%E7%AB%8B%E9%A2%84%E5%90%8E%E5%9B%A0%E5%AD%90"><span class="toc-number">6.</span> <span class="toc-text">2.5 筛选独立预后因子</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#2-5-1-LASSO-%E5%9B%9E%E5%BD%92%E5%88%86%E6%9E%90-x2F-%E9%99%8D%E7%BB%B4%E5%A4%84%E7%90%86"><span class="toc-number">6.1.</span> <span class="toc-text">2.5.1 LASSO 回归分析&#x2F;降维处理</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#2-5-2-%E4%B8%B4%E5%BA%8A%E8%B5%84%E6%96%99%E6%95%B4%E7%90%86"><span class="toc-number">6.2.</span> <span class="toc-text">2.5.2 临床资料整理</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#2-5-3-%E5%8D%95%E5%9B%A0%E7%B4%A0%E5%92%8C%E5%A4%9A%E5%9B%A0%E7%B4%A0-Cox-%E5%9B%9E%E5%BD%92%E5%88%86%E6%9E%90-x2F-%E7%A1%AE%E5%AE%9A%E7%8B%AC%E7%AB%8B%E9%A2%84%E5%90%8E%E5%9B%A0%E5%AD%90"><span class="toc-number">6.3.</span> <span class="toc-text">2.5.3 单因素和多因素 Cox 回归分析&#x2F;确定独立预后因子</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#2-5-4-%E8%A1%A8%E8%BE%BE%E9%87%8F%E7%BB%98%E5%88%B6%E5%88%97%E7%BA%BF%E5%9B%BE"><span class="toc-number">6.4.</span> <span class="toc-text">2.5.4 表达量绘制列线图</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2-6-%E9%AA%8C%E8%AF%81-SRC-%E5%9F%BA%E5%9B%A0%E5%9C%A8-HCC-%E4%B8%AD%E7%9A%84%E8%A1%A8%E8%BE%BE"><span class="toc-number">7.</span> <span class="toc-text">2.6 验证 SRC 基因在 HCC 中的表达</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2-7-ROC-%E6%9B%B2%E7%BA%BF%E5%88%86%E6%9E%90-SRC-%E5%9C%A8-HCC-%E6%82%A3%E8%80%85%E8%AF%8A%E6%96%AD%E5%8F%8A%E9%A2%84%E5%90%8E%E9%A2%84%E6%B5%8B%E4%B8%AD%E7%9A%84%E5%87%86%E7%A1%AE%E6%80%A7"><span class="toc-number">8.</span> <span class="toc-text">2.7 ROC 曲线分析 SRC 在 HCC 患者诊断及预后预测中的准确性</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E8%AE%A8%E8%AE%BA"><span class="toc-number"></span> <span class="toc-text">讨论</span></a></div></div><div class="author-info hide"><div class="author-info__avatar text-center"><img src="/img/avatar.png"></div><div class="author-info__name text-center">Yezi</div><div class="author-info__description text-center"></div><hr><div class="author-info-articles"><a class="author-info-articles__archives article-meta" href="/archives"><span class="pull-left">Articles</span><span class="pull-right">8</span></a><a class="author-info-articles__tags article-meta" href="/tags"><span class="pull-left">Tags</span><span class="pull-right">9</span></a><a class="author-info-articles__categories article-meta" href="/categories"><span class="pull-left">Categories</span><span class="pull-right">3</span></a></div></div></div><div id="content-outer"><div id="top-container" style="background-image: url(https://s2.loli.net/2022/03/04/IsybxLMGcDPW1kv.jpg)"><div id="page-header"><span class="pull-left"> <a id="site-name" href="/">Yezi Blog</a></span><i class="fa fa-bars toggle-menu pull-right" aria-hidden="true"></i><span class="pull-right menus">   <a class="site-page" href="/">Blog</a><a class="site-page" href="/archives">Posts</a><a class="site-page" href="/tags">Tags</a><a class="site-page" href="/categories">Categories</a></span><span class="pull-right"></span></div><div id="post-info"><div id="post-title">复现论文:铁死亡相关基因对肝细胞癌患者诊断及预后预测价值的研究</div><div id="post-meta"><time class="post-meta__date"><i class="fa fa-calendar" aria-hidden="true"></i> 2022-03-04</time><span class="post-meta__separator">|</span><i class="fa fa-inbox post-meta__icon" aria-hidden="true"></i><a class="post-meta__categories" href="/categories/Research/">Research</a></div></div></div><div class="layout" id="content-inner"><article id="post"><div class="article-container" id="post-content"><meta name="referrer" content="no-referrer" />
复现论文:铁死亡相关基因对肝细胞癌患者诊断及预后预测价值的研究

<h2 id="1-一般资料"><a href="#1-一般资料" class="headerlink" title="1. 一般资料"></a>1. 一般资料</h2><h4 id="1-1-tcga转录组测序数据下载"><a href="#1-1-tcga转录组测序数据下载" class="headerlink" title="1.1 tcga转录组测序数据下载"></a>1.1 tcga转录组测序数据下载</h4><span id="more"></span>
<p>下载链接:<br><a target="_blank" rel="noopener" href="https://portal.gdc.cancer.gov/repository?facetTab=files&filters=%7B%22op%22:%22and%22,%22content%22:%5B%7B%22op%22:%22in%22,%22content%22:%7B%22field%22:%22cases.primary_site%22,%22value%22:%5B%22liver%20and%20intrahepatic%20bile%20ducts%22%5D%7D%7D,%7B%22op%22:%22in%22,%22content%22:%7B%22field%22:%22cases.project.program.name%22,%22value%22:%5B%22TCGA%22%5D%7D%7D,%7B%22op%22:%22in%22,%22content%22:%7B%22field%22:%22cases.project.project_id%22,%22value%22:%5B%22TCGA-LIHC%22%5D%7D%7D,%7B%22op%22:%22in%22,%22content%22:%7B%22field%22:%22files.data_category%22,%22value%22:%5B%22transcriptome%20profiling%22%5D%7D%7D%5D%7D">tcga转录组测序数据下载</a><br>重点在下面的选项</p>
<ol>
<li>Gene Expression Quantification</li>
<li>HTSeq - Counts(视频中选取的是FPKM)</li>
</ol>
<p>Counts是原始数据,FPKM是矫正后数据,FPKM-UQ是另一种矫正后数据</p>
<blockquote>
<p>Counts是数据后台没有处理的原始表达量，而FPKM和FPKM-UQ是两种数据处理方法，也就是说，如果下载Counts数据，是表达量数据，如果下载FPKM数据，那么要注意这些数据是经过处理的。正常情况下，我们下载Counts数据就可以了，特殊情况选择FPKM数据也是可以的。<br><a target="_blank" rel="noopener" href="https://www.biowolf.cn/m/view.php?aid=138">https://www.biowolf.cn/m/view.php?aid=138</a><br>下载数据后，在数据分析时，用的方法也是不同的，Counts数据一般使用edgeR包或DESeq包，对数据做分析；如果下载FPKM数据，就不能使用edgeR包，只能只用DESeq包进行处理。<br>在使用edgeR包做Counts数据处理时，是需要对数据进行normalize的，所以我们在下载数据时，下载counts是比较常用的。</p>
</blockquote>
<p>之后依次下载cart文件,metadata文件(cart文件解压后自带manifest)</p>
<h4 id="1-2-tcga临床资料下载"><a href="#1-2-tcga临床资料下载" class="headerlink" title="1.2 tcga临床资料下载"></a>1.2 tcga临床资料下载</h4><p>点击下载链接:  <a target="_blank" rel="noopener" href="https://portal.gdc.cancer.gov/repository?facetTab=files&filters=%7B%22op%22:%22and%22,%22content%22:%5B%7B%22op%22:%22in%22,%22content%22:%7B%22field%22:%22cases.primary_site%22,%22value%22:%5B%22liver%20and%20intrahepatic%20bile%20ducts%22%5D%7D%7D,%7B%22op%22:%22in%22,%22content%22:%7B%22field%22:%22cases.project.program.name%22,%22value%22:%5B%22TCGA%22%5D%7D%7D,%7B%22op%22:%22in%22,%22content%22:%7B%22field%22:%22cases.project.project_id%22,%22value%22:%5B%22TCGA-LIHC%22%5D%7D%7D,%7B%22op%22:%22in%22,%22content%22:%7B%22field%22:%22files.data_category%22,%22value%22:%5B%22transcriptome%20profiling%22%5D%7D%7D%5D%7D">tcga转录组测序数据下载</a>,依次选择</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Data Category-clinical</span><br><span class="line">Clinical Supplement</span><br><span class="line">bcr xml</span><br></pre></td></tr></table></figure>
<p>之后依次下载manifest文件,cart文件,metadata文件,clinic文件<br>​</p>
<h2 id="2-1-铁死亡相关差异表达基因的筛选"><a href="#2-1-铁死亡相关差异表达基因的筛选" class="headerlink" title="2.1 铁死亡相关差异表达基因的筛选"></a>2.1 铁死亡相关差异表达基因的筛选</h2><h3 id="2-1-筛选差异基因"><a href="#2-1-筛选差异基因" class="headerlink" title="2.1 筛选差异基因"></a>2.1 筛选差异基因</h3><p>目标:<strong>从 TCGA 数据库下载 371 例患者肿瘤组织和 50 例患者癌旁组织的转录组数据，即为包括 60 483 个基因的表达谱，剔除无表达数据的基因，剩余 60 432 个基因。</strong></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br></pre></td><td class="code"><pre><span class="line">install.packages(c(&quot;rjson&quot;,&quot;jsonio&quot;,&quot;jsonlite&quot;)) #未安装包则先安装</span><br><span class="line">library(rjson)</span><br><span class="line">library(jsonlite)</span><br><span class="line">library(stringr)</span><br><span class="line">library(biomaRt)</span><br><span class="line">library(dplyr)</span><br><span class="line">rm(list = ls())</span><br><span class="line">setwd(&quot;/Users/sandy/Downloads/test/1.一般资料/1.1. tcga转录组测序数据下载&quot;) # 需修改</span><br><span class="line"></span><br><span class="line">json &lt;- fromJSON(txt = list.files(pattern = &quot;.json&quot;))</span><br><span class="line">MMRF_ID&lt;-list()</span><br><span class="line">matrix_MMRF&lt;-list()</span><br><span class="line">filename_in_json&lt;-list()</span><br><span class="line">for (i in 1:nrow(json)) &#123;</span><br><span class="line">  MMRF_ID[i]&lt;-json[[3]][[i]]</span><br><span class="line">&#125;</span><br><span class="line">MMRF_ID&lt;-do.call(rbind, MMRF_ID)</span><br><span class="line">for (i in 1:nrow(json)) &#123;</span><br><span class="line">  filename_in_json[i]&lt;-json[i,4]</span><br><span class="line">&#125;</span><br><span class="line">filename_in_json&lt;-do.call(rbind, filename_in_json)</span><br><span class="line">filename_in_json&lt;-as.character(filename_in_json)</span><br><span class="line">for (i in 1:length(filename_in_json)) &#123;</span><br><span class="line">  matrix_MMRF[[i]] = read.table(gzfile(filename_in_json[i]), header = F)[,2]</span><br><span class="line">&#125;</span><br><span class="line">probematrix&lt;-read.table( gzfile(filename_in_json[1]), header = F)[,1]</span><br><span class="line"></span><br><span class="line">matrix_MMRF&lt;-do.call(cbind, matrix_MMRF)</span><br><span class="line">matrix_MMRF&lt;-cbind(probematrix,matrix_MMRF)</span><br><span class="line"></span><br><span class="line">colnames(matrix_MMRF)&lt;-c(&quot;ID&quot;,MMRF_ID[,1])</span><br><span class="line"></span><br><span class="line">#可先进行id转换,但为与论文步骤保持一致留待下一步</span><br><span class="line">#ID转换</span><br><span class="line">#matrix_MMRF&lt;-read.table(&quot;symbol.txt&quot;,sep = &quot;\t&quot;,header = T,check.names = F)</span><br><span class="line">matrix_MMRF[,1]&lt;-str_remove(matrix_MMRF[,1],pattern = &quot;[.][0-9]*&quot;)</span><br><span class="line">ensembl = useMart(biomart=&quot;ensembl&quot;,dataset=&quot;hsapiens_gene_ensembl&quot;)</span><br><span class="line">geneENS&lt;- getBM(attributes=c(&quot;ensembl_gene_id&quot;,&quot;hgnc_symbol&quot;),</span><br><span class="line">                filters = &quot;ensembl_gene_id&quot;,</span><br><span class="line">                values=matrix_MMRF[,1],</span><br><span class="line">                mart=ensembl)</span><br><span class="line">all&lt;-merge(geneENS,matrix_MMRF,by.x = &quot;ensembl_gene_id&quot;,by.y = &quot;ID&quot;,all=T)</span><br><span class="line">for (i in 1:nrow(all))&#123;</span><br><span class="line">  #i &lt;- 1</span><br><span class="line">  if (is.na(all[,2][i])|all[,2][i]==&quot;&quot;) &#123;</span><br><span class="line">    all[,2][i]&lt;-all[,1][i]</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line">all&lt;-all[,-1]</span><br><span class="line">all&lt;-all[all2$hgnc_symbol!=&quot;&quot;,]</span><br><span class="line">setwd(&quot;../&quot;)</span><br><span class="line">write.table(all,&quot;symbol_未分组.txt&quot;,sep = &quot;\t&quot;,row.names = F,col.names = T)</span><br><span class="line"></span><br><span class="line">#重新排序,将癌旁排在前面便于下一步筛选</span><br><span class="line">all &lt;- matrix_MMRF</span><br><span class="line">iris &lt;- tibble::as_tibble(all)</span><br><span class="line">#0-9为癌数据,排在后面</span><br><span class="line">out &lt;- select(iris, -matches(&quot;.-0[1-9][A-Z]-.&quot;), everything())</span><br><span class="line">setwd(&quot;../&quot;)</span><br><span class="line">write.table(out,&quot;ensl_分组.txt&quot;,sep = &quot;\t&quot;,row.names = F,col.names = T)</span><br><span class="line"></span><br><span class="line">#名称修改</span><br><span class="line">#MM_col&lt;-str_replace(MMRF_ID[,1],pattern = &quot;aq-&quot;,replacement = &quot;&quot;)</span><br><span class="line">#MM_col&lt;-str_replace(MM_col,pattern = &quot;R&quot;,replacement = &quot;&quot;)</span><br><span class="line">colnames(out)&lt;-str_sub(colnames(out),1,12)</span><br><span class="line">write.table(out,&quot;ensl_order.txt&quot;,sep = &quot;\t&quot;,row.names = F,col.names = T)</span><br><span class="line"></span><br><span class="line">#str(matrix_MMRF)</span><br><span class="line">#str(MMRF_ID)</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p>得到括 60 483 个基因的表达量表格.<br>​</p>
<p>目标:<strong>剔除无表达数据的基因，剩余 60 432 个基因。</strong>待筛选差异基因步骤继续.<br><strong>采用“Limma”包筛选这些基因中的差异基因，得到 4 983 个上调差异基因和 1 865 个下调差异基因 （图 1A）。</strong><br><img src="https://cdn.nlark.com/yuque/0/2022/png/23203300/1645426616337-0c254ba9-40c2-4fc5-940e-c1e5797d9a05.png#clientId=u610a49c9-185d-4&crop=0&crop=0&crop=1&crop=1&from=paste&height=371&id=u7e31c66f&margin=%5Bobject%20Object%5D&name=image.png&originHeight=371&originWidth=896&originalType=binary&ratio=1&rotation=0&showTitle=false&size=232039&status=done&style=none&taskId=ueaf1fff0-d7fb-4c25-a0c7-a0497f59e5b&title=&width=896" alt="image.png"></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line">rm(list = ls())</span><br><span class="line">library(limma)</span><br><span class="line">inputFile=&quot;/Users/sandy/Downloads/test/1.一般资料/ensl_order.txt&quot;</span><br><span class="line">pValuefilter=0.05       </span><br><span class="line">log2FCfilter=2              </span><br><span class="line">conNum=50                   </span><br><span class="line">treatNum=374  </span><br><span class="line">setwd(&quot;/Users/sandy/Downloads/test/2.1 筛选差异基因&quot;)  </span><br><span class="line"></span><br><span class="line">outTab=data.frame()</span><br><span class="line">grade=c(rep(1,conNum),rep(2,treatNum))</span><br><span class="line">rt=read.table(inputFile,sep=&quot;\t&quot;,header=T,check.names=F)</span><br><span class="line">rt=as.matrix(rt)</span><br><span class="line">rownames(rt)=rt[,1]</span><br><span class="line">exp=rt[,2:ncol(rt)]</span><br><span class="line">dimnames=list(rownames(exp),colnames(exp))</span><br><span class="line">data=matrix(as.numeric(as.matrix(exp)),nrow=nrow(exp),dimnames=dimnames)</span><br><span class="line">#剔除无表达数据的基因</span><br><span class="line">data=avereps(data)</span><br><span class="line">data=data[rowMeans(data)&gt;0,]</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p>剔除无表达数据的基因，剩余 56,454 个基因。我也不知道为什么不一样..总之下一步吧<br><strong>采用“Limma”包筛选这些基因中的差异基因，得到 4 983 个上调差异基因和 1 865 个下调差异基因 （图 1A）。</strong></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br></pre></td><td class="code"><pre><span class="line">for(i in row.names(data))&#123;</span><br><span class="line">  #i &lt;- row.names(data)[363]</span><br><span class="line">	geneName=unlist(strsplit(i,&quot;\\|&quot;,))[1]</span><br><span class="line">	geneName=gsub(&quot;\\/&quot;, &quot;_&quot;, geneName)</span><br><span class="line">	rt=rbind(expression=data[i,],grade=grade)</span><br><span class="line">	rt=as.matrix(t(rt))</span><br><span class="line">	wilcoxTest=wilcox.test(expression ~ grade, data=rt)</span><br><span class="line">	conGeneMeans=mean(data[i,1:conNum])</span><br><span class="line">	treatGeneMeans=mean(data[i,(conNum+1):ncol(data)])</span><br><span class="line">	log2FC=log2(treatGeneMeans)-log2(conGeneMeans)</span><br><span class="line">	pvalue=wilcoxTest$p.value</span><br><span class="line">	conMed=median(data[i,1:conNum])</span><br><span class="line">	treatMed=median(data[i,(conNum+1):ncol(data)])</span><br><span class="line">	diffMed=treatMed-conMed</span><br><span class="line">	if( ((log2FC&gt;0) &amp; (diffMed&gt;0)) | ((log2FC&lt;0) &amp; (diffMed&lt;0)) )&#123;  </span><br><span class="line">		outTab=rbind(outTab,cbind(gene=i,conMean=conGeneMeans,treatMean=treatGeneMeans,log2FC=log2FC,pValue=pvalue))</span><br><span class="line">	&#125;</span><br><span class="line">&#125;</span><br><span class="line">pValue=outTab[,&quot;pValue&quot;]</span><br><span class="line">fdr=p.adjust(as.numeric(as.vector(pValue)),method=&quot;fdr&quot;)</span><br><span class="line">outTab=cbind(outTab,fdr=fdr)</span><br><span class="line"></span><br><span class="line">#有意义的基因</span><br><span class="line">write.table(outTab,file=&quot;tcga.all.xls&quot;,sep=&quot;\t&quot;,row.names=F,quote=F)</span><br><span class="line">#符合筛选的基因</span><br><span class="line">outDiff=outTab[( abs(as.numeric(as.vector(outTab$log2FC)))&gt;log2FCfilter &amp; as.numeric(as.vector(outTab$pValue))&lt;pValuefilter),]</span><br><span class="line">write.table(outDiff,file=&quot;tcga.diff.xls&quot;,sep=&quot;\t&quot;,row.names=F,quote=F)</span><br><span class="line">write.table(outDiff,file=&quot;tcga.diff.txt&quot;,sep=&quot;\t&quot;,row.names=F,quote=F)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">heatmap=rbind(ID=colnames(data[as.vector(outDiff[,1]),]),data[as.vector(outDiff[,1]),])</span><br><span class="line">write.table(heatmap,file=&quot;tcga.diffExp.txt&quot;,sep=&quot;\t&quot;,col.names=F,quote=F)</span><br></pre></td></tr></table></figure>
<p>得到 <code>22,924</code>个有意义的基因.<br>得到了<code>3,273</code>个满足筛选的基因,</p>
<h3 id="2-1-2-火山图"><a href="#2-1-2-火山图" class="headerlink" title="2.1.2 火山图"></a>2.1.2 火山图</h3><p>得到了一些差异基因,用来做火山图</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br></pre></td><td class="code"><pre><span class="line">rm(list = ls())</span><br><span class="line"># 安装ggpubr和ggthemes包</span><br><span class="line">#install.packages(&quot;ggpubr&quot;)</span><br><span class="line">#install.packages(&quot;ggthemes&quot;)</span><br><span class="line"># 加载ggpubr和ggthemes包</span><br><span class="line">library(ggplot2)</span><br><span class="line">library(ggpubr)</span><br><span class="line">library(ggthemes)</span><br><span class="line"></span><br><span class="line"># 读取数据</span><br><span class="line">deg.data &lt;- read.table(&quot;/Users/sandy/Downloads/text/1.2.1 差异基因分析/01.Limma包筛选差异基因/tcga.all.xls&quot;,header=TRUE,check.names =F, sep = &quot;\t&quot;)</span><br><span class="line">colnames(deg.data)[1] &lt;- &quot;ID&quot;</span><br><span class="line"># 显示文件</span><br><span class="line">head(deg.data)</span><br><span class="line"></span><br><span class="line"># 对差异pvalue进行log10转换</span><br><span class="line">deg.data$logQ &lt;- -log10(deg.data$pValue)</span><br><span class="line"></span><br><span class="line"># 绘制基本热图</span><br><span class="line">ggscatter(deg.data, x = &quot;log2FC&quot;, y = &quot;logQ&quot;) + theme_base()</span><br><span class="line"></span><br><span class="line"># 新加一列Group</span><br><span class="line">deg.data$Group = &quot;normal&quot;</span><br><span class="line"># 将adj.P.Val小于0.05，logFC大于2</span><br><span class="line"># 将adj.P.Val小于0.05，logFC小于2的基因设为显著下调基因</span><br><span class="line">deg.data$Group[which( (deg.data$pValue &lt; 0.05) &amp; (deg.data$log2FC &gt; 2) )] = &quot;up&quot;</span><br><span class="line">deg.data$Group[which( (deg.data$pValue &lt; 0.05) &amp; (deg.data$log2FC &lt; -2) )] = &quot;down&quot;</span><br><span class="line"># 查看上调和下调基因数</span><br><span class="line">table(deg.data$Group)</span><br><span class="line"></span><br><span class="line"># 绘制新的火山图</span><br><span class="line">ggscatter(deg.data, x = &quot;log2FC&quot;, y = &quot;logQ&quot;,</span><br><span class="line">          color = &quot;Group&quot;) + theme_base()</span><br><span class="line"></span><br><span class="line"># 改变火山图颜色(palette)和点的大小(size)</span><br><span class="line">ggscatter(deg.data, x = &quot;log2FC&quot;, y = &quot;logQ&quot;,</span><br><span class="line">          color = &quot;Group&quot;, </span><br><span class="line">          palette = c(&quot;green&quot;, &quot;gray&quot;, &quot;red&quot;),</span><br><span class="line">          size = 1) + theme_base()</span><br><span class="line"></span><br><span class="line"># 为火山图添加logP分界线条(geom_hline)和logFC分界线条(geom_vline)</span><br><span class="line">ggscatter(deg.data, x = &quot;log2FC&quot;, y = &quot;logQ&quot;,</span><br><span class="line">          color = &quot;Group&quot;, </span><br><span class="line">          palette = c(&quot;green&quot;, &quot;gray&quot;, &quot;red&quot;),</span><br><span class="line">          size = 1) + theme_base() + </span><br><span class="line">  geom_hline(yintercept = 1.30, linetype=&quot;dashed&quot;) +</span><br><span class="line">  geom_vline(xintercept = c(-2,2), linetype=&quot;dashed&quot;)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"># 新加一列Label</span><br><span class="line">deg.data$Label = &quot;&quot;</span><br><span class="line"># 对差异表达基因的log2FC值进行从大到小排序取top 20</span><br><span class="line">deg.data &lt;- deg.data[order(abs(deg.data$log2FC),decreasing = T), ]</span><br><span class="line">log2FC.genes &lt;- head(deg.data$ID, 20)</span><br><span class="line">#head(deg.data)</span><br><span class="line"># 对差异表达基因的p值进行从小到大排top 20</span><br><span class="line"># deg.data &lt;- deg.data[order(deg.data$logQ)]</span><br><span class="line"># 高表达的基因中，选择FDR最小的20</span><br><span class="line">deg.data &lt;- deg.data[order(abs(deg.data$logQ),decreasing = T), ]</span><br><span class="line">fdr.genes &lt;- head(deg.data$ID, 20)</span><br><span class="line"></span><br><span class="line"># 将log2FC.genes 和fdr.genes合并，并加入到Label</span><br><span class="line">deg.top20.genes &lt;- c(as.character(log2FC.genes), as.character(fdr.genes))</span><br><span class="line">deg.data$Label[match(deg.top20.genes, deg.data$ID)] &lt;- deg.top20.genes</span><br><span class="line">#print (deg.data$Label)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"># 改变火山图点的颜色和坐标轴标注，使图片更美观</span><br><span class="line">ggscatter(deg.data, x = &quot;log2FC&quot;, y = &quot;logQ&quot;,</span><br><span class="line">          color = &quot;Group&quot;, </span><br><span class="line">          palette = c(&quot;#00BA38&quot;, &quot;#BBBBBB&quot;, &quot;#F8766D&quot;),</span><br><span class="line">          size = 2,</span><br><span class="line">          label = deg.data$Label, </span><br><span class="line">          font.label = 8, </span><br><span class="line">          repel = T,</span><br><span class="line">          xlab = &quot;log2FC&quot;, </span><br><span class="line">          ylab = &quot;-log10(p value)&quot;) + </span><br><span class="line">  theme_base() + </span><br><span class="line">  geom_hline(yintercept = 1.30, linetype=&quot;dashed&quot;) +</span><br><span class="line">  geom_vline(xintercept = c(-2,2), linetype=&quot;dashed&quot;)</span><br></pre></td></tr></table></figure>
<p><strong>另外，从FerrDb 数据库下载 123 个铁死亡标记基因、150 个铁死亡驱动基因和 109 个铁死亡抑制基因。采用“venn”包对差异基因和铁死亡基因取交集，获得 40 个铁死亡相关上调差异基因和 31 个铁死亡相关下调差异基因（图 1B），共 71 个铁死亡相关差异表达基因。</strong></p>
<h3 id="2-1-3-Venn图"><a href="#2-1-3-Venn图" class="headerlink" title="2.1.3 Venn图"></a>2.1.3 Venn图</h3><p>进入老版本FerrDB下载所需数据,为了读取方便下载scv格式数据(新版有txt),地址:<br><a target="_blank" rel="noopener" href="http://www.zhounan.org/ferrdb/legacy/operations/download.html">http://www.zhounan.org/ferrdb/legacy/operations/download.html</a><br><img src="https://cdn.nlark.com/yuque/0/2022/png/23203300/1645432316648-9233ea15-e5f0-4d5c-9a0e-2610e56b2b4e.png#clientId=u610a49c9-185d-4&crop=0&crop=0&crop=1&crop=1&from=paste&height=467&id=u296d4d25&margin=%5Bobject%20Object%5D&name=image.png&originHeight=467&originWidth=1228&originalType=binary&ratio=1&rotation=0&showTitle=false&size=135926&status=done&style=none&taskId=u60f583c6-07e1-443b-8eda-9490232aace&title=&width=1228" alt="image.png"></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br></pre></td><td class="code"><pre><span class="line">rm(list=ls())</span><br><span class="line">setwd(&quot;/Users/sandy/Downloads/test/2.1.3 FerrDB下载所需数据&quot;)</span><br><span class="line">install.packages(&quot;VennDiagram&quot;)</span><br><span class="line">library(VennDiagram)</span><br><span class="line">library(readr)</span><br><span class="line">library(dplyr)</span><br><span class="line">csvFileNames   &lt;-  dir(path = &quot;./&quot;, pattern = &quot;csv$&quot;)</span><br><span class="line">#readr::read_csv返回的是tibble_list,而read.csv返回character,谨记</span><br><span class="line">FromecsvGenedata   &lt;-  lapply(csvFileNames ,</span><br><span class="line">                              function(x)&#123;            # x = csvFileNames[1]</span><br><span class="line">                                result   &lt;-  read.csv(paste( file = x, sep=&quot;&quot;))</span><br><span class="line">                                csvdataframe &lt;- result[,1]</span><br><span class="line">                                return(csvdataframe)&#125;)</span><br><span class="line">names(FromecsvGenedata) = csvFileNames</span><br><span class="line"></span><br><span class="line">#以下方法可以得到三个基因的总数据框,但不利于后续处理,废弃</span><br><span class="line">#vennFromCsv &lt;- data.frame()</span><br><span class="line">#vennFromCsv &lt;- do.call(cbind, lapply(lapply(FromecsvGenedata, unlist), `length&lt;-`, max(lengths(FromecsvGenedata))))</span><br><span class="line">#str(vennFromCsv)</span><br><span class="line"></span><br><span class="line">driver &lt;- FromecsvGenedata[[1]]</span><br><span class="line">marker &lt;- FromecsvGenedata[[2]]</span><br><span class="line">suppressor &lt;- FromecsvGenedata[[3]]</span><br><span class="line"></span><br><span class="line">diff &lt;- read.table(&quot;/Users/sandy/Downloads/test/2.1 筛选差异基因/tcga.diff.xls&quot;,header = T,sep = &quot;\t&quot;)</span><br><span class="line"></span><br><span class="line">#ID转换</span><br><span class="line">library(biomaRt)</span><br><span class="line">#注意ensb号的.后可能有两位数的数字,仅使用[1-9]不足以排除,加上*为选取0-n个[1-9]字符</span><br><span class="line">diff[,1] &lt;- str_remove(diff[,1],pattern = &quot;[.][0-9]*&quot;)</span><br><span class="line">ensm &lt;- diff[,1]</span><br><span class="line">ensembl = useMart(biomart=&quot;ensembl&quot;,dataset=&quot;hsapiens_gene_ensembl&quot;)</span><br><span class="line">geneENS&lt;- getBM(attributes=c(&quot;ensembl_gene_id&quot;,&quot;hgnc_symbol&quot;),</span><br><span class="line">                filters = &quot;ensembl_gene_id&quot;,</span><br><span class="line">                values=ensm,</span><br><span class="line">                mart=ensembl)</span><br><span class="line">colnames(diff)[1] &lt;- colnames(geneENS)[1]</span><br><span class="line">deg.data &lt;- left_join(diff,geneENS)</span><br><span class="line">deg.data &lt;- deg.data[is.na(deg.data$hgnc_symbol)==F,]</span><br><span class="line">deg.data &lt;- deg.data[(deg.data$hgnc_symbol)!=&quot;&quot;,]</span><br><span class="line">up.diff &lt;- deg.data[which( (deg.data$pValue &lt; 0.05) &amp; (deg.data$log2FC &gt; 2) ),][,7]</span><br><span class="line">down.diff &lt;- deg.data[which( (deg.data$pValue &lt; 0.05) &amp; (deg.data$log2FC &lt; -2) ),][,7]</span><br><span class="line"></span><br><span class="line">grid.newpage();</span><br><span class="line">venn.plot &lt;- venn.diagram(</span><br><span class="line">  x = list(Set1=up.diff,Set2=marker,Set3=suppressor,Set4=marker,Set5=down.diff),</span><br><span class="line">  filename = NULL,</span><br><span class="line">  col = &quot;black&quot;,</span><br><span class="line">  fill = c(&quot;dodgerblue&quot;, &quot;goldenrod1&quot;, &quot;darkorange1&quot;, &quot;seagreen3&quot;, &quot;orchid3&quot;),</span><br><span class="line">  alpha = 0.50,</span><br><span class="line">  cex = c(1.5, 1.5, 1.5, 1.5, 1.5, 1, 0.8, 1, 0.8, 1, 0.8, 1, 0.8,</span><br><span class="line">          1, 0.8, 1, 0.55, 1, 0.55, 1, 0.55, 1, 0.55, 1, 0.55, 1, 1, 1, 1, 1, 1.5),</span><br><span class="line">  cat.col = c(&quot;dodgerblue&quot;, &quot;goldenrod1&quot;, &quot;darkorange1&quot;, &quot;seagreen3&quot;, &quot;orchid3&quot;),</span><br><span class="line">  cat.cex = 1.5,</span><br><span class="line">  cat.fontface = &quot;bold&quot;,</span><br><span class="line">  margin = 0.05</span><br><span class="line">);</span><br><span class="line">grid.draw(venn.plot);</span><br><span class="line">grid.newpage();</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p>使用VennDiagram包作图,很遗憾还是完全不对,但是没啥办法,这里得到的结果是上调基因15个,下调1个(FPKM)<br><img src="https://cdn.nlark.com/yuque/0/2022/png/23203300/1645437211644-30295219-034e-4202-8e53-c229dd2fdfd0.png#clientId=ue0f12bd7-bef7-4&crop=0&crop=0&crop=1&crop=1&from=paste&height=455&id=u576d3810&margin=%5Bobject%20Object%5D&name=image.png&originHeight=455&originWidth=572&originalType=binary&ratio=1&rotation=0&showTitle=false&size=95852&status=done&style=none&taskId=u80d97ebe-c8ce-433f-a8b2-9ceda9f29b4&title=&width=572" alt="image.png"><br>counts: up:2316-2303&#x3D;13,down: 133-131&#x3D;2.<br><img src="https://cdn.nlark.com/yuque/0/2022/png/23203300/1646127535634-febb5375-8951-4015-9f6a-d35e7f5fd8e3.png#clientId=u219538ee-b92c-4&crop=0&crop=0&crop=1&crop=1&from=paste&height=551&id=u71856084&margin=%5Bobject%20Object%5D&name=image.png&originHeight=1102&originWidth=1350&originalType=binary&ratio=1&rotation=0&showTitle=false&size=287960&status=done&style=none&taskId=u44179a6d-e08e-4ff6-903e-07dc15c06e9&title=&width=675" alt="image.png"></p>
<h2 id="2-2-模块基因筛选"><a href="#2-2-模块基因筛选" class="headerlink" title="2.2 模块基因筛选"></a>2.2 模块基因筛选</h2><p>将 71 个铁死亡相关差异表达基因输入 STRING 网站展示基因与基因之间的相互作用关系，剔除没有任何相互作用关系的基因。将基因相互作用网络图输入 Cytoscape软件，采用 MCODE 插件进行模块化分析（图 2A），选取评分最高（评分&#x3D;30.64）的模块为后续研究对象，包括 15 个基因（图 2B），这些基因即为模块基因。</p>
<h2 id="2-3-生存相关模块基因筛选"><a href="#2-3-生存相关模块基因筛选" class="headerlink" title="2.3 生存相关模块基因筛选"></a>2.3 生存相关模块基因筛选</h2><p>采用 Kaplan-Meier Plotter 网站分析 15 个模块基因分别与 OS 和 PFS 的关系。其中，11 个基因与 HCC 患者的OS 存在显著相关性（P＜0.05，表 1），9 个基因与 HCC 患者的 PFS存在显著相关性（P＜0.05，表 2）。选取与患者 OS和 PFS 均具有显著相关性的基因，包括 ALB、CDKN2A、HRAS、JUN、MAPK3、NOX4 和 SRC 这 7 个基因，这些基因即为生存相关模块基因。<br><img src="https://cdn.nlark.com/yuque/0/2022/png/23203300/1645884760965-ab50c5e7-c64c-482d-b88e-497385579b8d.png#clientId=ua037c222-cdf6-4&crop=0&crop=0&crop=1&crop=1&from=paste&height=377&id=u994a764e&margin=%5Bobject%20Object%5D&name=image.png&originHeight=377&originWidth=887&originalType=binary&ratio=1&rotation=0&showTitle=false&size=327385&status=done&style=none&taskId=u868e537d-5cdd-4ede-ab26-960af3e0370&title=&width=887" alt="image.png"></p>
<h2 id="2-4-生存相关模块基因的功能富集分析"><a href="#2-4-生存相关模块基因的功能富集分析" class="headerlink" title="2.4 生存相关模块基因的功能富集分析"></a>2.4 生存相关模块基因的功能富集分析</h2><p>采用 DAVID 软件对 7 个生存相关模块基因进行功能富集分析，“ggplot2”包对富集结果进行可视化。KEGG 分析结果发现，这 7 个基因主要富集在 ErbB 信号通路、GnRH信号通路、VEGF 信号通路、AGE-RAGE 信号通路以及线粒体自噬等通路中（表3）。GO 富集分析主要包括细胞组成（CC）、分子功能（MF）和生物过程（BP）三个部分。结果发现，这 7个基因主要参与细胞核、细胞膜内腔、细胞器内腔、细胞与基因连接等的构成，主要的分子功能为结合蛋白，参与的主要生物学过程为细胞程序性死亡过程。此外，将这 7 个基因输入 Cytoscape 软件的 ClueGO+CluePedia 插件进行基因与KEGG 通路之间的相互作用分析，结果发现，除 ALB 外其余 6 个基因均与上述 KEGG 通路具有多重相互作用关系，尤其是 CDKN2A、HRAS、JUN、MAPK3 和 SRC 基因。</p>
<h2 id="2-5-筛选独立预后因子"><a href="#2-5-筛选独立预后因子" class="headerlink" title="2.5 筛选独立预后因子"></a>2.5 筛选独立预后因子</h2><p>采用 LASSO 回归分析对 7 个生存相关模块基因进行降维处理，结果得到 CDKN2A、HRAS、JUN、MAPK3 和SRC这 5 个基因（图 3A、B），即功能富集分析处于核心位置的5 个基因。采用单因素和多因素 Cox 回归分析进一步确定独立预后因子，同时纳入年龄、性别、T 分期、TNM分期和临床分级等临床资料 ，结果显示 ，SRC 基 因可以作为一个独立的预后预测指标（HR&#x3D;1.19319，95%CI为 1.00326~1.41908，P&#x3D;0.04585）。根据 SRC 基因的表达量绘制列线图，预测患者 1、3、5 年的总体生存可能性，结果显示，SRC 表达越高，患者生存可能性越小（图 3C）。校准曲线的虚线为对角线，表示列线图的结果可信（图 3D）。<br><img src="https://cdn.nlark.com/yuque/0/2022/png/23203300/1645885220562-40801b91-60ec-4405-8233-5a6ce74ff030.png#clientId=ua037c222-cdf6-4&crop=0&crop=0&crop=1&crop=1&from=paste&height=360&id=u3cda8797&margin=%5Bobject%20Object%5D&name=image.png&originHeight=360&originWidth=505&originalType=binary&ratio=1&rotation=0&showTitle=false&size=47243&status=done&style=none&taskId=ua07fdb85-1436-4539-8d9f-9cdac8dfa11&title=&width=505" alt="image.png"><br><img src="https://cdn.nlark.com/yuque/0/2022/png/23203300/1645885236575-4b5cd044-e707-4a8e-9c21-a05af4760637.png#clientId=ua037c222-cdf6-4&crop=0&crop=0&crop=1&crop=1&from=paste&height=636&id=u6688efd2&margin=%5Bobject%20Object%5D&name=image.png&originHeight=636&originWidth=980&originalType=binary&ratio=1&rotation=0&showTitle=false&size=177734&status=done&style=none&taskId=ue821d36f-fdda-448f-a7e1-591fc467599&title=&width=980" alt="image.png"></p>
<h3 id="2-5-1-LASSO-回归分析-x2F-降维处理"><a href="#2-5-1-LASSO-回归分析-x2F-降维处理" class="headerlink" title="2.5.1 LASSO 回归分析&#x2F;降维处理"></a>2.5.1 LASSO 回归分析&#x2F;降维处理</h3><h3 id="2-5-2-临床资料整理"><a href="#2-5-2-临床资料整理" class="headerlink" title="2.5.2 临床资料整理"></a>2.5.2 临床资料整理</h3><p><strong>同时纳入年龄、性别、T 分期、TNM分期和临床分级等临床资料 ，</strong><br>下载clinical.cart.2022-03-01.tar解压,得到clinical.tsv,点开可以查看一下信息.</p>
<table>
<thead>
<tr>
<th>case_submitter_id</th>
<th>gender</th>
<th>ajcc_pathologic_t</th>
</tr>
</thead>
<tbody><tr>
<td>TCGA-CC-A8HU</td>
<td>female</td>
<td>T3</td>
</tr>
<tr>
<td>ajcc_pathologic_m</td>
<td>ajcc_pathologic_n</td>
<td>ajcc_pathologic_stage</td>
</tr>
<tr>
<td>M0</td>
<td>N0</td>
<td>Stage IIIA</td>
</tr>
<tr>
<td>vital_status</td>
<td></td>
<td></td>
</tr>
<tr>
<td>Dead</td>
<td></td>
<td></td>
</tr>
</tbody></table>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">rm(list=ls())</span><br><span class="line">cli &lt;- read.table(&quot;/Users/sandy/Downloads/test/1.一般资料/02.tcga临床资料下载/clinical.cart.2022-03-01/clinical.tsv&quot;,sep = &quot;\t&quot;,quote = &quot;&quot;,fill = T,header = T)</span><br><span class="line">clinic &lt;- subset(cli[,c(&quot;case_submitter_id&quot;,&quot;days_to_death&quot;,&quot;gender&quot;,&quot;ajcc_pathologic_t&quot;,&quot;ajcc_pathologic_m&quot;,&quot;ajcc_pathologic_n&quot;,&quot;ajcc_pathologic_stage&quot;,&quot;vital_status&quot;)])</span><br><span class="line">clinic$fustat = &quot;0&quot;</span><br><span class="line">clinic$fustat[clinic$vital_status==&quot;Dead&quot;] = &quot;1&quot;</span><br><span class="line">clinic &lt;- clinic[duplicated(clinic),]</span><br><span class="line">write.table(clinic,&quot;/Users/sandy/Downloads/test/1.一般资料/02.tcga临床资料下载/clinic.txt&quot;,col.names = T,row.names = F,quote = F,sep = &quot;\t&quot;)</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h3 id="2-5-3-单因素和多因素-Cox-回归分析-x2F-确定独立预后因子"><a href="#2-5-3-单因素和多因素-Cox-回归分析-x2F-确定独立预后因子" class="headerlink" title="2.5.3 单因素和多因素 Cox 回归分析&#x2F;确定独立预后因子"></a>2.5.3 单因素和多因素 Cox 回归分析&#x2F;确定独立预后因子</h3><p>结果显示，SRC 基因可以作为一个独立的预后预测指标（HR&#x3D;1.19319，95%CI为 1.00326~1.41908，P&#x3D;0.04585）。</p>
<h3 id="2-5-4-表达量绘制列线图"><a href="#2-5-4-表达量绘制列线图" class="headerlink" title="2.5.4 表达量绘制列线图"></a>2.5.4 表达量绘制列线图</h3><p>根据 SRC 基因的表达量绘制列线图，预测患者 1、3、5 年的总体生存可能性，结果显示，SRC 表达越高，患者生存可能性越小（图 3C）。校准曲线的虚线为对角线，表示列线图的结果可信（图 3D）。</p>
<h2 id="2-6-验证-SRC-基因在-HCC-中的表达"><a href="#2-6-验证-SRC-基因在-HCC-中的表达" class="headerlink" title="2.6 验证 SRC 基因在 HCC 中的表达"></a>2.6 验证 SRC 基因在 HCC 中的表达</h2><p>进一步验证 SRC 基因在 HCC 中的表达情况。首先，基于CCLE 数据集（<a target="_blank" rel="noopener" href="https://portals.broadinstitute.org/ccle/about">https://portals.broadinstitute.org/ccle/about</a>）下载 21 个肝癌细胞系 SRC 基因的表达，采用“ggplot2”包绘制图像，结果发现，SRC 基因在全部肝癌细胞系中表达上调（图 4A）。进一步对 HCC 组织和癌旁组织非配对样本中 SRC 进行表达分析，结果显示，与癌旁组织比较，SRC 在HCC 组织中显著高表达（P&lt;0.001，图 4B）。配对样本中也得到了类似的结果，HCC 组织中 SRC 显著高表达（P&lt;0.001， 图 4C）。这些结果均表明 SRC 在 HCC 中高表达。<br><img src="https://cdn.nlark.com/yuque/0/2022/png/23203300/1645885806438-33a7e01d-a2ca-42fd-b265-d59962e0be0b.png#clientId=ua037c222-cdf6-4&crop=0&crop=0&crop=1&crop=1&from=paste&height=578&id=u776acd83&margin=%5Bobject%20Object%5D&name=image.png&originHeight=578&originWidth=977&originalType=binary&ratio=1&rotation=0&showTitle=false&size=250963&status=done&style=none&taskId=u2f84712b-7edc-4df3-88fc-e9acb6883b7&title=&width=977" alt="image.png"><br>注：图 A 为 SRC 基因在肝癌细胞系中的表达；图 B 为 SRC 基因在 HCC 组织和癌旁组织非配对样本中的表达；图 C 为 SRC 基因在 HCC 组织和癌旁组织配对样本中的表达。</p>
<h2 id="2-7-ROC-曲线分析-SRC-在-HCC-患者诊断及预后预测中的准确性"><a href="#2-7-ROC-曲线分析-SRC-在-HCC-患者诊断及预后预测中的准确性" class="headerlink" title="2.7 ROC 曲线分析 SRC 在 HCC 患者诊断及预后预测中的准确性"></a>2.7 ROC 曲线分析 SRC 在 HCC 患者诊断及预后预测中的准确性</h2><p>采用 ROC 曲线分析 SRC 在 HCC 患者诊断及预后预测中的准确性。SRC 基因在 HCC 患者诊断中具有较高 的 准 确 性（AUC &#x3D;0.784，95% CI 为 0.739 ~0.829，图5A），同时在 HCC 患者 1、3、5 年预后预测中也具有较高的准确性（AUC1 年&#x3D;0.706，AUC3 年&#x3D;0.607，AUC5 年&#x3D;0.594， 图 5B）。<br><img src="https://cdn.nlark.com/yuque/0/2022/png/23203300/1645885817770-f49ac8a1-eaac-488f-96cf-2c0a3dbe3730.png#clientId=ua037c222-cdf6-4&crop=0&crop=0&crop=1&crop=1&from=paste&height=321&id=u216f0ed6&margin=%5Bobject%20Object%5D&name=image.png&originHeight=321&originWidth=700&originalType=binary&ratio=1&rotation=0&showTitle=false&size=114963&status=done&style=none&taskId=u64a39472-7b2a-4aaf-aba7-0ad077ec7bc&title=&width=700" alt="image.png"><br>注：图 A 为 ROC 曲线分析 SRC 在 HCC 患者诊断中的准确性；图 B 为 ROC 曲线分析 SRC 在 HCC 患者预后预测中的准确性。</p>
<h1 id="讨论"><a href="#讨论" class="headerlink" title="讨论"></a>讨论</h1><p> 诱导细胞铁死亡是一种新兴的癌症治疗方法。近年来，多项研究显示，促进癌细胞对铁死亡的敏感性有助于癌症治疗[7-8]。由于铁死亡可调控癌细胞增殖，有助于治疗癌症，由此推测铁死亡相关基因可能在预后预测方面也发挥了一定作用。为此，本研究首先通过生物信息学系统分析了 382 个铁死亡相关基因在 HCC 组织中的表达情况，并逐步筛选出与 HCC 患者诊断和预后评估相关的基因。结果发现，铁死亡抑制因子 SRC 可以作为 HCC 的一个独立预后预测指标（P&#x3D;0.04585）。SRC 表达越高，患者生存预后越差。</p>
</div></article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">Author: </span><span class="post-copyright-info"><a href="mailto:undefined">Yezi</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">Link: </span><span class="post-copyright-info"><a href="http://example.com/2022/03/04/铁死亡论文复现_1/">http://example.com/2022/03/04/铁死亡论文复现_1/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">Copyright Notice: </span><span class="post-copyright-info">All articles in this blog are licensed under <a target="_blank" rel="noopener" href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a> unless stating additionally.</span></div></div><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/Iron-death/">Iron death</a></div><nav id="pagination"><div class="prev-post pull-left"><a href="/2022/03/09/%E8%AE%BA%E6%96%87%E9%98%85%E8%AF%BB_%E9%BB%84%E8%BF%9E%E8%A7%A3%E6%AF%92%E6%B1%A4%E6%8A%97%E6%8A%91%E9%83%81%E6%B4%BB%E6%80%A7%E7%9A%84%E7%BD%91%E7%BB%9C%E8%8D%AF%E7%90%86%E5%AD%A6%E5%92%8C%E4%BB%A3%E8%B0%A2%E7%BB%84%E5%AD%A6%E5%88%86%E6%9E%90/"><i class="fa fa-chevron-left">  </i><span>论文阅读_黄连解毒汤抗抑郁活性的网络药理学和代谢组学分析</span></a></div><div class="next-post pull-right"><a href="/2022/02/20/markdown%E8%AF%AD%E6%B3%95/"><span>markdown语法</span><i class="fa fa-chevron-right"></i></a></div></nav><div id="vcomment"></div><script src="https://cdn1.lncld.net/static/js/3.0.4/av-min.js"></script><script src="https://cdn.jsdelivr.net/npm/valine/dist/Valine.min.js"></script><script>var notify = 'ture' == 'true';
var verify = 'false' == 'true';
var record_ip = '' == 'true';
var GUEST_INFO = ['nick','mail','link'];
var guest_info = 'nick,mail,link'.split(',').filter(function(item){
  return GUEST_INFO.indexOf(item) > -1
});
guest_info = guest_info.length == 0 ? GUEST_INFO :guest_info;
window.valine = new Valine({
  el:'#vcomment',
  notify:notify,
  verify:verify,
  recordIP:record_ip,
  appId:'WCwykmLMAzeUjkLJXSOPUxsw-gzGzoHsz',
  appKey:'lrcStLJI87jyaFnRQj1szybX',
  placeholder:'Just go go',
  avatar:'mm',
  guest_info:guest_info,
  pageSize:'10',
  lang: 'zh-cn'
})</script></div></div><footer class="footer-bg" style="background-image: url(https://s2.loli.net/2022/03/04/IsybxLMGcDPW1kv.jpg)"><div class="layout" id="footer"><div class="copyright">&copy;2022 By Yezi</div><div class="framework-info"><span>Driven - </span><a target="_blank" rel="noopener" href="http://hexo.io"><span>Hexo</span></a><span class="footer-separator">|</span><span>Theme - </span><a target="_blank" rel="noopener" href="https://github.com/Molunerfinn/hexo-theme-melody"><span>Melody</span></a></div><div class="footer_custom_text">hitokoto</div><div class="busuanzi"><script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script><span id="busuanzi_container_page_pv"><i class="fa fa-file"></i><span id="busuanzi_value_page_pv"></span><span></span></span></div></div></footer><i class="fa fa-arrow-up" id="go-up" aria-hidden="true"></i><script src="https://cdn.jsdelivr.net/npm/animejs@latest/anime.min.js"></script><script src="https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.js"></script><script src="https://cdn.jsdelivr.net/npm/velocity-animate@latest/velocity.min.js"></script><script src="https://cdn.jsdelivr.net/npm/velocity-ui-pack@latest/velocity.ui.min.js"></script><script src="/js/utils.js?version=1.9.0"></script><script src="/js/fancybox.js?version=1.9.0"></script><script src="/js/sidebar.js?version=1.9.0"></script><script src="/js/copy.js?version=1.9.0"></script><script src="/js/fireworks.js?version=1.9.0"></script><script src="/js/transition.js?version=1.9.0"></script><script src="/js/scroll.js?version=1.9.0"></script><script src="/js/head.js?version=1.9.0"></script><script>if(/Android|webOS|iPhone|iPod|iPad|BlackBerry/i.test(navigator.userAgent)) {
  $('#nav').addClass('is-mobile')
  $('footer').addClass('is-mobile')
  $('#top-container').addClass('is-mobile')
}</script></body></html>